<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        
        
        <link rel="shortcut icon" href="../../../img/favicon.ico">

	<title>Method Background - LSCC Documentation</title>

        <link href="../../../css/bootstrap-custom.min.css" rel="stylesheet">
        <link href="../../../css/font-awesome-4.0.3.css" rel="stylesheet">
        <link rel="stylesheet" href="../../../css/highlight.css">
        <link href="../../../css/base.css" rel="stylesheet">
        <link href="../../../includes/media/mdx_showable.css" rel="stylesheet">
        <link href="../../../includes/media/mdx_showable_user.css" rel="stylesheet">
        <link href="../../../includes/media/slides.css" rel="stylesheet">
        <link href="../../../includes/media/tute.css" rel="stylesheet">

        <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
        <!--[if lt IE 9]>
            <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
            <script src="https://oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
        <![endif]-->

        
    </head>

    <body>

        <div class="navbar navbar-default navbar-fixed-top" role="navigation">
    <div class="container">

        <!-- Collapsed navigation -->
        <div class="navbar-header">
            <!-- Expander button -->
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>

            <!-- Main title -->
            <a class="navbar-brand" href="../../..">LSCC Documentation</a>
        </div>

        <!-- Expanded navigation -->
        <div class="navbar-collapse collapse">
            <!-- Main navigation -->
            <ul class="nav navbar-nav">
            
            
                <li >
                    <a href="../../..">Home</a>
                </li>
            
            
            
                <li class="dropdown active">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Tutorials <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li >
    <a href="../../">Overview</a>
</li>

                    
                        
<li >
    <a href="../../ngs_overview/NGS_Overview/">NGS Overview</a>
</li>

                    
                        
<li >
    <a href="../../using_git/Using_Git/">Using Git and Github for revision control</a>
</li>

                    
                        
<li >
    <a href="../../gvl_launch/gvl_launch/">Launching a Personal GVL Server</a>
</li>

                    
                        
<li >
    <a href="../../galaxy_101/galaxy_101/">Introduction to Galaxy</a>
</li>

                    
                        
<li >
    <a href="../../galaxy-workflows/galaxy-workflows/">Galaxy Workflows</a>
</li>

                    
                        
<li >
    <a href="../../rna_seq_dge_basic/rna_seq_basic_background/">RNA-seq Basic Background</a>
</li>

                    
                        
<li >
    <a href="../../rna_seq_dge_basic/rna_seq_basic_tutorial/">RNA-seq Basic</a>
</li>

                    
                        
<li >
    <a href="../../rna_seq_dge_advanced/rna_seq_advanced_tutorial/">RNA-seq Advanced</a>
</li>

                    
                        
<li >
    <a href="../../rna_seq_exp_design/rna_seq_experimental_design/">RNA-seq Experimental Design</a>
</li>

                    
                        
<li >
    <a href="../../hpc/">Introduction to HPC</a>
</li>

                    
                        
<li >
    <a href="../../unix/">Introduction to Unix</a>
</li>

                    
                        
<li >
    <a href="../../python_overview/python_overview/">Python Overview</a>
</li>

                    
                        
  <li class="dropdown-submenu">
    <a tabindex="-1" href="">Microbial de novo Assembly for Illumina Data</a>
    <ul class="dropdown-menu">
        
            
<li >
    <a href="../assembly/">Tutorial</a>
</li>

        
            
<li class="active">
    <a href="./">Method Background</a>
</li>

        
    </ul>
  </li>

                    
                        
<li >
    <a href="../../variant_calling_galaxy_1/variant_calling_galaxy_1/">Introduction to Variant Calling using Galaxy</a>
</li>

                    
                        
  <li class="dropdown-submenu">
    <a tabindex="-1" href="">Variant Detection Advanced</a>
    <ul class="dropdown-menu">
        
            
<li >
    <a href="../../var_detect_advanced/var_detect_advanced/">Var detect advanced</a>
</li>

        
            
<li >
    <a href="../../var_detect_advanced/var_detect_advanced_background/">Var detect advanced background</a>
</li>

        
    </ul>
  </li>

                    
                        
  <li class="dropdown-submenu">
    <a tabindex="-1" href="">Identifying proteins from mass spectrometry data</a>
    <ul class="dropdown-menu">
        
            
<li >
    <a href="../../proteomics_basic/proteomics_basic/">Tutorial</a>
</li>

        
            
<li >
    <a href="../../proteomics_basic/background_galaxy/">Galaxy Background</a>
</li>

        
            
<li >
    <a href="../../proteomics_basic/background_data_formats/">Data Formats Background</a>
</li>

        
            
<li >
    <a href="../../proteomics_basic/background_findcountitems_workflow/">Create find count items workflow</a>
</li>

        
            
<li >
    <a href="../../proteomics_basic/background_protein_databases/">Protein Databases</a>
</li>

        
            
<li >
    <a href="../../proteomics_basic/background_search_engines/">Search Engines</a>
</li>

        
            
<li >
    <a href="../../proteomics_basic/background_protein_prophet/">Protein Prophet</a>
</li>

        
            
<li >
    <a href="../../alignment/alignment/">Alignment Tutorial</a>
</li>

        
    </ul>
  </li>

                    
                        
<li >
    <a href="../../genomespace/genomespace/">Introduction to GenomeSpace</a>
</li>

                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Protocols <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li >
    <a href="../assembly-protocol/">De Novo Genome Assembly for Illumina Data</a>
</li>

                    
                    </ul>
                </li>
            
            
            </ul>

            <!-- Search, Navigation and Repo links -->
            <ul class="nav navbar-nav navbar-right">
                <li>
                    <a href="#" data-toggle="modal" data-target="#mkdocs_search_modal">
                        <i class="fa fa-search"></i> Search
                    </a>
                </li>
                <li >
                    <a rel="next" href="../assembly/">
                        <i class="fa fa-arrow-left"></i> Previous
                    </a>
                </li>
                <li >
                    <a rel="prev" href="../../variant_calling_galaxy_1/variant_calling_galaxy_1/">
                        Next <i class="fa fa-arrow-right"></i>
                    </a>
                </li>
                
            </ul>
        </div>
    </div>
</div>

        <div class="container">
            <div class="col-md-3"><div class="bs-sidebar hidden-print affix well" role="complementary">
    <ul class="nav bs-sidenav">
    
        <li class="main active"><a href="#de-novo-genome-assembly-using-velvet">De novo genome assembly using Velvet</a></li>
        
            <li><a href="#background">Background</a></li>
        
            <li><a href="#the-galaxy-workflow-platform">The Galaxy workflow platform</a></li>
        
            <li><a href="#de-novo-assembly-with-velvet-and-the-velvet-optimiser">De novo assembly with Velvet and the Velvet Optimiser.</a></li>
        
            <li><a href="#references">References</a></li>
        
    
    </ul>
</div></div>
            <div class="col-md-9" role="main">

<h1 id="de-novo-genome-assembly-using-velvet">De novo genome assembly using Velvet</h1>
<h2 id="background">Background</h2>
<h3 id="introduction-to-de-novo-assembly">Introduction to de novo assembly</h3>
<p>DNA sequence assembly from short fragments (&lt; 200 bp) is often the first step of any bioinformatic analysis. The goal of assembly is to take the millions of short reads produced by sequencing instruments and re-construct the DNA from which the reads originated.</p>
<p>The sequence assembly issue was neatly summed up by the following quote:</p>
<p><em>"The problem of sequence assembly can be compared to taking many copies of a book, passing them all through a shredder, and piecing a copy of the book back together from only shredded pieces.  The book may have many repeated paragraphs,  and some shreds may be modified to have typos.  Excerpts from another book may be added in, and some shreds may be completely unrecognizable."</em> – Wikipedia: Sequence assembly.</p>
<p>An addition to the above for paired end sequencing is that now some of the shreds are quite long but only about 10% of the words from both ends of the shred are known.</p>
<p>This tutorial describes de novo assembly of Illumina short reads using the Velvet assembler (Zerbino  et al. 2008, 2009) and the Velvet Optimiser (Gladman &amp; Seemann, 2009) from within the Galaxy workflow management system.</p>
<hr />
<h2 id="the-galaxy-workflow-platform">The Galaxy workflow platform</h2>
<h3 id="what-is-galaxy">What is Galaxy?</h3>
<p>Galaxy is an online bioinformatics workflow management system. Essentially, you upload your files, create various analysis pipelines and run them, then visualise your results.</p>
<p>Galaxy is really an interface to the various tools that do the data processing; each of these tools could be run from the command line, outside of Galaxy. Galaxy makes it easier to link up the tools together and visualise the entire analysis pipeline.</p>
<p>Galaxy uses the concept of 'histories'. Histories are sets of data and workflows that act on that data. The data for this workshop is available in a shared history, which you can import into your own Galaxy account</p>
<p>Learn more about Galaxy here</p>
<p><strong>Figure 1: The Galaxy interface</strong></p>
<p>Tools on the left, data in the middle, analysis workflow on the right.</p>
<p><img src="../media/screenshot-fastqc-output.png" width=100% /></p>
<hr />
<h2 id="de-novo-assembly-with-velvet-and-the-velvet-optimiser">De novo assembly with Velvet and the Velvet Optimiser.</h2>
<h3 id="velvet">Velvet</h3>
<p>Velvet is software to perform dna assembly from short reads by manipulating de Bruijn graphs. It is capable of forming long contigs (n50 of in excess of 150kb) from paired end short reads. It has several input parameters for controlling the structure of the de Bruijn graph and these must be set optimally to get the best assembly possible. Velvet can read Fasta, FastQ, sam or bam files. However, it ignores any quality scores and simply relies on sequencing depth to resolve errors. The Velvet Optimiser software performs many Velvet assemblies with various parameter sets and searches for the optimal assembly automatically.</p>
<h3 id="de-bruijn-graphs">de Bruijn graphs</h3>
<p>A de Bruijn graph is a directed graph which represents overlaps between sequences of symbols. The size of the sequence contained in the nodes of the graph is called the word-length or k-mer size. In Figure 2, the word length is 3. The two symbols are 1 and 0. Each node in the graph has the last two symbols of the previous node and 1 new symbol. Sequences of symbols can be produced by traversing the graph and adding the “new” symbol to the growing sequence.</p>
<p><strong>Figure 2: A de Bruijn graph of word length 3 for the symbols 1 and 0.</strong></p>
<p><img src="../media/debruijn.jpg" width=80% /></p>
<p><em>From: https://cameroncounts.wordpress.com/2015/02/28/1247/</em></p>
<p>Velvet constructs a de Bruijn graph of the reads. It has 4 symbols (A, C, G and T - N’s are converted to A’s) The word length (or k-mer size) is one of Velvet’s prime parameters.</p>
<p>Velvet is not the only assembly software that works in this manner. Euler, Edena and SOAP de novo are examples of others.</p>
<h3 id="the-velvet-algorithm">The Velvet algorithm</h3>
<h4 id="step-1-hashing-the-reads"><em>Step 1: Hashing the reads.</em></h4>
<ul>
<li>Velvet breaks up each read into k-mers of length k.</li>
<li>A k-mer is a k length subsequence of the read.</li>
<li>A 36 base pair long read would have 6 different 31-mers.</li>
<li>The k-mers and their reverse complements are added to a hash table to categorize them.</li>
<li>Each k-mer is stored once but the number of times it appears is also recorded.</li>
<li>This step is performed by “velveth” - one of the programs in the Velvet suite.</li>
</ul>
<h4 id="step-2-constructing-the-de-bruijn-graph"><em>Step 2: Constructing the de Bruijn graph.</em></h4>
<ul>
<li>Velvet adds the k-mers one-by-one to the graph.</li>
<li>Adjacent k-mers overlap by k-1 nucleotides.</li>
<li>A k-mer which has no k-1 overlaps with any k-mer already on the graph starts a new node.</li>
<li>Each node stores the average number of times its k-mers appear in the hash table.</li>
<li>Figure 3 shows a section of a de Bruijn graph constructed by Velvet for k=5.</li>
<li>Different sequences can be read off the graph by following a different path through it. (Figure 3)</li>
</ul>
<p><strong>Figure 3: Section of a simple de Bruijn graph of reads with k-mer size 5. Coloured sequences are constructed by following the appropriately coloured line through the graph.</strong> <em>(Base figure Zerbino et al 2008.)</em></p>
<p><img src="../media/velvet-graph.png" width=80% /></p>
<h4 id="step-3-simplification-of-the-graph"><em>Step 3: Simplification of the graph.</em></h4>
<ul>
<li>Chain merging: When there are two connected nodes in the graph without a divergence, merge the two nodes.</li>
<li>Tip clipping: Tips are short (typically) chains of nodes that are disconnected on one end. They will be clipped if their length is &lt; 2 x k or their average k-mer depth is much less than the continuing path.</li>
<li>Bubble removal: Bubbles are redundant paths that start and end at the same nodes (Figure 4.) They are created by sequencing errors, biological variants or slightly varying repeat sequences.</li>
<li>Velvet compares the paths using dynamic programming.</li>
<li>If they are highly similar, the paths are merged.</li>
<li>Error removal: Erroneous connections are removed by using a “coverage cutoff”. Genuine short nodes which cannot be simplified should have a high coverage. An attempt is made to resolve repeats using the “expected coverage” of the graph nodes.</li>
<li>Paired end read information: Velvet uses algorithms called “Pebble” and “Rock Band” (Zerbino et al 2009) to order the nodes with respect to one another in order to scaffold them into longer contigs.</li>
</ul>
<p><strong>Figure 4: Representation of “bubbles” in a Velvet de Bruijn graph.</strong>  <em>(Base figure Zerbino et al 2008.)</em></p>
<p><img src="../media/velvet-bubbles.png" width=80% /></p>
<h4 id="step-4-read-off-the-contigs"><em>Step 4: Read off the contigs.</em></h4>
<ul>
<li>Follow the chains of nodes through the graph and “read off” the bases to create the contigs.</li>
<li>Where there is an ambiguous divergence/convergence, stop the current contig and start a new one.</li>
</ul>
<h3 id="k-mer-size-and-coverage-cutoff-values">K-mer size and coverage cutoff values</h3>
<p>The size of the k-mers that construct the graph is very important and has a large effect on the outcome of the assembly. Generally, small k-mers create a graph with increased connectivity, more ambiguity (more divergences) and less clear “paths” through the graph. Large k-mers produce graphs with less connectivity but higher specificity. The paths through the graph are clearer but they are less connected and prone to breaking down.</p>
<p>The coverage cutoff c used during the error correction step of Velvet also has a significant effect on the output of the assembly process. If c is too low, the assembly will contain nodes of the graph that are the product of sequencing errors and misconnections. If c is too high, it can create mis-assemblies in the contigs and destroys lots of useful data.</p>
<p>Each dataset has its own optimum values for the k-mer size and the coverage cutoff used in the error removal step. Choosing them appropriately is one of the challenges faced by new users of the Velvet software.</p>
<h3 id="velvet-optimiser">Velvet Optimiser</h3>
<p>The Velvet Optimiser chooses the optimal values for k and c automatically by performing many runs of Velvet (partially in parallel) and interrogating the subsequent assemblies.  It uses different optimisation functions for k and c and these can be user controlled.</p>
<p>It requires the user to input a range of k values to search (to cut down on running time).</p>
<h2 id="references">References</h2>
<p>http://en.wikipedia.org/wiki/Sequence_assembly</p>
<p>Zerbino DR, Birney E, Velvet: algorithms for de novo short read assembly using de Bruijn graphs, Genome Research, 2008, 18:821-829</p>
<p>Zerbino DR, McEwen GK, Margulies EH, Birney E, Pebble and rock band: heuristic resolution of repeats and scaffolding in the velvet short-read de novo assembler. PLoS One. 2009; 4(12):e8407.</p>
<p>Gladman SL, Seemann T, Velvet Optimiser, http://www.vicbioinformatics.com/software.shtml 2009.</p></div>
        </div>

        <footer class="col-md-12">
            <hr>
            
            <center>Documentation built with <a href="http://www.mkdocs.org/">MkDocs</a>.</center>
        </footer>

        <script src="../../../js/jquery-1.10.2.min.js"></script>
        <script src="../../../js/bootstrap-3.0.3.min.js"></script>
        <script src="../../../js/highlight.pack.js"></script>
        <script>var base_url = '../../..';</script>
        <script data-main="../../../mkdocs/js/search.js" src="../../../mkdocs/js/require.js"></script>
        <script src="../../../js/base.js"></script>
        <script src="../../../includes/media/delayed-jquery.js"></script>
        <script src="../../../includes/media/mdx_showable.js"></script>

        <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="Search Modal" aria-hidden="true">
            <div class="modal-dialog">
                <div class="modal-content">
                    <div class="modal-header">
                        <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
                        <h4 class="modal-title" id="exampleModalLabel">Search</h4>
                    </div>
                    <div class="modal-body">
                        <p>
                            From here you can search these documents. Enter
                            your search terms below.
                        </p>
                        <form role="form">
                            <div class="form-group">
                                <input type="text" class="form-control" placeholder="Search..." id="mkdocs-search-query">
                            </div>
                        </form>
                        <div id="mkdocs-search-results"></div>
                    </div>
                    <div class="modal-footer">
                    </div>
                </div>
            </div>
        </div>
    </body>
</html>