



<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
      
      
      
        <meta name="lang:clipboard.copy" content="Copy to clipboard">
      
        <meta name="lang:clipboard.copied" content="Copied to clipboard">
      
        <meta name="lang:search.language" content="en">
      
        <meta name="lang:search.pipeline.stopwords" content="True">
      
        <meta name="lang:search.pipeline.trimmer" content="True">
      
        <meta name="lang:search.result.none" content="No matching documents">
      
        <meta name="lang:search.result.one" content="1 matching document">
      
        <meta name="lang:search.result.other" content="# matching documents">
      
        <meta name="lang:search.tokenizer" content="[\s\-]+">
      
      <link rel="shortcut icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.0.4, mkdocs-material-4.4.0">
    
    
      
        <title>de novo assembly of PacBio reads (unix) - Bioinformatics Documentation</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/application.0284f74d.css">
      
      
    
    
      <script src="../../assets/javascripts/modernizr.74668098.js"></script>
    
    
      
        <link href="https://fonts.gstatic.com" rel="preconnect" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono&display=fallback">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
    
    <link rel="stylesheet" href="../../assets/fonts/material-icons.css">
    
    
      <link rel="stylesheet" href="../../extra.css">
    
    
<script async src="https://www.googletagmanager.com/gtag/js?id=G-EMNLJD0DWP"></script>
<script>
    window.dataLayer = window.dataLayer || [];
    function gtag() {
        dataLayer.push(arguments);
    }
    gtag('js', new Date());
    gtag('config', 'G-EMNLJD0DWP');
</script>

    
  </head>
  
    <body dir="ltr">
  
    <svg class="md-svg">
      <defs>
        
        
          <svg xmlns="http://www.w3.org/2000/svg" width="416" height="448" viewBox="0 0 416 448" id="__github"><path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19T128 352t-18.125-8.5-10.75-19T96 304t3.125-20.5 10.75-19T128 256t18.125 8.5 10.75 19T160 304zm160 0q0 10-3.125 20.5t-10.75 19T288 352t-18.125-8.5-10.75-19T256 304t3.125-20.5 10.75-19T288 256t18.125 8.5 10.75 19T320 304zm40 0q0-30-17.25-51T296 232q-10.25 0-48.75 5.25Q229.5 240 208 240t-39.25-2.75Q130.75 232 120 232q-29.5 0-46.75 21T56 304q0 22 8 38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0 37.25-1.75t35-7.375 30.5-15 20.25-25.75T360 304zm56-44q0 51.75-15.25 82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5T212 416q-19.5 0-35.5-.75t-36.875-3.125-38.125-7.5-34.25-12.875T37 371.5t-21.5-28.75Q0 312 0 260q0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25 30.875Q171.5 96 212 96q37 0 70 8 26.25-20.5 46.75-30.25T376 64q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34 99.5z"/></svg>
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" data-md-component="overlay" for="__drawer"></label>
    
      <a href="#long-read-assembly-workshop" tabindex="1" class="md-skip">
        Skip to content
      </a>
    
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        <a href="../.." title="Bioinformatics Documentation" class="md-header-nav__button md-logo">
          
            <i class="md-icon"></i>
          
        </a>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="__drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <div class="md-flex__ellipsis md-header-nav__title" data-md-component="title">
          
            <span class="md-header-nav__topic">
              Bioinformatics Documentation
            </span>
            <span class="md-header-nav__topic">
              
                de novo assembly of PacBio reads (unix)
              
            </span>
          
        </div>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <label class="md-icon md-icon--search md-header-nav__button" for="__search"></label>
          
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query" data-md-state="active">
      <label class="md-icon md-search__icon" for="__search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset" tabindex="-1">
        &#xE5CD;
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
        
      </div>
      
        <div class="md-flex__cell md-flex__cell--shrink">
          <div class="md-header-nav__source">
            


  

<a href="https://github.com/melbournebioinformatics/MelBioInf_docs/" title="Go to repository" class="md-source" data-md-source="github">
  
    <div class="md-source__icon">
      <svg viewBox="0 0 24 24" width="24" height="24">
        <use xlink:href="#__github" width="24" height="24"></use>
      </svg>
    </div>
  
  <div class="md-source__repository">
    melbournebioinformatics/MelBioInf_docs
  </div>
</a>
          </div>
        </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
        
      
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="__drawer">
    <a href="../.." title="Bioinformatics Documentation" class="md-nav__button md-logo">
      
        <i class="md-icon"></i>
      
    </a>
    Bioinformatics Documentation
  </label>
  
    <div class="md-nav__source">
      


  

<a href="https://github.com/melbournebioinformatics/MelBioInf_docs/" title="Go to repository" class="md-source" data-md-source="github">
  
    <div class="md-source__icon">
      <svg viewBox="0 0 24 24" width="24" height="24">
        <use xlink:href="#__github" width="24" height="24"></use>
      </svg>
    </div>
  
  <div class="md-source__repository">
    melbournebioinformatics/MelBioInf_docs
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href="../.." title="Home" class="md-nav__link">
      Home
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-2" type="checkbox" id="nav-2">
    
    <label class="md-nav__link" for="nav-2">
      Tools and skill development
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-2">
        Tools and skill development
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../guides/bioinfo/" title="What is Bioinformatics?" class="md-nav__link">
      What is Bioinformatics?
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../guides/galaxy/" title="Introduction to the Galaxy Platform" class="md-nav__link">
      Introduction to the Galaxy Platform
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../galaxy-workflows/galaxy-workflows/" title="Introduction to Galaxy Workflows" class="md-nav__link">
      Introduction to Galaxy Workflows
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../python_overview/python_overview/" title="Introduction to Python" class="md-nav__link">
      Introduction to Python
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../using_git/Using_Git/" title="Introduction to Git and Github" class="md-nav__link">
      Introduction to Git and Github
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../hpc/hpc/" title="Introduction to HPC" class="md-nav__link">
      Introduction to HPC
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../unix/unix/" title="Introduction to Unix" class="md-nav__link">
      Introduction to Unix
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../intro_R_biologists/intro_R_biologists/" title="Introduction to R for Biologists" class="md-nav__link">
      Introduction to R for Biologists
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../Genome_browsers/GenomeBrowsers_Intro/" title="Introduction to Genome Browsers" class="md-nav__link">
      Introduction to Genome Browsers
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../rna_seq_exp_design/rna_seq_experimental_design/" title="RNA-seq Experimental Design" class="md-nav__link">
      RNA-seq Experimental Design
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../docker/docker/" title="Containerized Bioinformatics" class="md-nav__link">
      Containerized Bioinformatics
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../cwl/cwl/" title="Common Workflow Language for Bioinformatics" class="md-nav__link">
      Common Workflow Language for Bioinformatics
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../bionitio/bionitio/" title="Command-line software development" class="md-nav__link">
      Command-line software development
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-3" type="checkbox" id="nav-3" checked>
    
    <label class="md-nav__link" for="nav-3">
      Genomics
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-3">
        Genomics
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-3-1" type="checkbox" id="nav-3-1">
    
    <label class="md-nav__link" for="nav-3-1">
      Variant detection
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="2">
      <label class="md-nav__title" for="nav-3-1">
        Variant detection
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../var_detect_advanced/var_detect_advanced_background/" title="Introduction to Variant detection" class="md-nav__link">
      Introduction to Variant detection
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../variant_calling_galaxy_1/variant_calling_galaxy_1/" title="Variant Calling part 1 (Galaxy)" class="md-nav__link">
      Variant Calling part 1 (Galaxy)
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../var_detect_advanced/var_detect_advanced/" title="Variant Calling part 2 (Galaxy)" class="md-nav__link">
      Variant Calling part 2 (Galaxy)
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../longread_sv_calling/longread_sv_calling/" title="Long-read Structural Variant Calling" class="md-nav__link">
      Long-read Structural Variant Calling
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-3-2" type="checkbox" id="nav-3-2" checked>
    
    <label class="md-nav__link" for="nav-3-2">
      de novo assembly
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="2">
      <label class="md-nav__title" for="nav-3-2">
        de novo assembly
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../assembly/assembly-background/" title="Introduction to de novo assembly with Velvet" class="md-nav__link">
      Introduction to de novo assembly with Velvet
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../assembly/assembly-protocol/" title="Introduction to de novo genome assembly for Illumina reads" class="md-nav__link">
      Introduction to de novo genome assembly for Illumina reads
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../assembly/assembly/" title="de novo assembly of Illumina reads using Velvet (Galaxy)" class="md-nav__link">
      de novo assembly of Illumina reads using Velvet (Galaxy)
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../assembly/spades/" title="de novo assembly of Illumina reads using Spades (Galaxy)" class="md-nav__link">
      de novo assembly of Illumina reads using Spades (Galaxy)
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="__toc">
    
      
    
    
      <label class="md-nav__link md-nav__link--active" for="__toc">
        de novo assembly of PacBio reads (unix)
      </label>
    
    <a href="./" title="de novo assembly of PacBio reads (unix)" class="md-nav__link md-nav__link--active">
      de novo assembly of PacBio reads (unix)
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#overview" title="Overview" class="md-nav__link">
    Overview
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#1-get-started" title="1. Get started" class="md-nav__link">
    1. Get started
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#mac-users" title="Mac users" class="md-nav__link">
    Mac users
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#windows-users" title="Windows users" class="md-nav__link">
    Windows users
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#activate-the-conda-environment" title="Activate the conda environment" class="md-nav__link">
    Activate the conda environment
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#create-a-new-working-directory-on-your-remote-computer" title="Create a new working directory on your remote computer." class="md-nav__link">
    Create a new working directory on your remote computer.
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#2-get-data" title="2. Get data" class="md-nav__link">
    2. Get data
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#3-assemble" title="3. Assemble" class="md-nav__link">
    3. Assemble
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#4-check-assembly-output" title="4. Check assembly output" class="md-nav__link">
    4. Check assembly output
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#questions" title="Questions" class="md-nav__link">
    Questions
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#5-trim-and-circularise" title="5. Trim and circularise" class="md-nav__link">
    5. Trim and circularise
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#run-circlator" title="Run Circlator" class="md-nav__link">
    Run Circlator
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#check-the-output" title="Check the output" class="md-nav__link">
    Check the output
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#questions_1" title="Questions" class="md-nav__link">
    Questions
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#6-find-smaller-plasmids" title="6. Find smaller plasmids" class="md-nav__link">
    6. Find smaller plasmids
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#align-illumina-reads-to-the-pacbio-contig" title="Align Illumina reads to the PacBio contig" class="md-nav__link">
    Align Illumina reads to the PacBio contig
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#extract-unmapped-illumina-reads" title="Extract unmapped Illumina reads" class="md-nav__link">
    Extract unmapped Illumina reads
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#assemble-the-unmapped-reads" title="Assemble the unmapped reads" class="md-nav__link">
    Assemble the unmapped reads
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#trim-the-plasmid" title="Trim the plasmid" class="md-nav__link">
    Trim the plasmid
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#collect-contigs" title="Collect contigs" class="md-nav__link">
    Collect contigs
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#questions_2" title="Questions" class="md-nav__link">
    Questions
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#7-correct-the-assembly" title="7. Correct the assembly" class="md-nav__link">
    7. Correct the assembly
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#make-an-alignment-file" title="Make an alignment file" class="md-nav__link">
    Make an alignment file
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#run-pilon" title="Run Pilon" class="md-nav__link">
    Run Pilon
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#questions_3" title="Questions" class="md-nav__link">
    Questions
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#8-comparative-genomics" title="8. Comparative Genomics" class="md-nav__link">
    8. Comparative Genomics
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#assemblies" title="Assemblies" class="md-nav__link">
    Assemblies
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#annotations" title="Annotations" class="md-nav__link">
    Annotations
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#9-summary" title="9. Summary" class="md-nav__link">
    9. Summary
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
    
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../hybrid_assembly/nanopore_assembly/" title="Hybrid genome assembly - Nanopore and Illumina" class="md-nav__link">
      Hybrid genome assembly - Nanopore and Illumina
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-4" type="checkbox" id="nav-4">
    
    <label class="md-nav__link" for="nav-4">
      Transcriptomics
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-4">
        Transcriptomics
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../rna_seq_exp_design/rna_seq_experimental_design/" title="RNAseq Experimental Design" class="md-nav__link">
      RNAseq Experimental Design
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-4-2" type="checkbox" id="nav-4-2">
    
    <label class="md-nav__link" for="nav-4-2">
      Bulk RNAseq
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="2">
      <label class="md-nav__title" for="nav-4-2">
        Bulk RNAseq
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../rna_seq_dge_basic/rna_seq_basic_background/" title="Introduction to bulk RNAseq analysis" class="md-nav__link">
      Introduction to bulk RNAseq analysis
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../rna_seq_dge_basic/rna_seq_basic_tutorial/" title="RNAseq analysis using HISAT2 (Galaxy)" class="md-nav__link">
      RNAseq analysis using HISAT2 (Galaxy)
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../rna_seq_dge_basic/rna_seq_basic_tuxedo/" title="RNAseq analysis using Tuxedo (Galaxy)" class="md-nav__link">
      RNAseq analysis using Tuxedo (Galaxy)
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../rna_seq_dge_advanced/rna_seq_advanced_tutorial/" title="RNAseq differential expression tool comparision (Galaxy)" class="md-nav__link">
      RNAseq differential expression tool comparision (Galaxy)
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../rna_seq_dge_in_r/rna_seq_r/" title="RNAseq differential expression in R" class="md-nav__link">
      RNAseq differential expression in R
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

        
          
          
          


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-4-3" type="checkbox" id="nav-4-3">
    
    <label class="md-nav__link" for="nav-4-3">
      Singlecell RNAseq
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="2">
      <label class="md-nav__title" for="nav-4-3">
        Singlecell RNAseq
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../singlecell/" title="10X single-cell RNA-seq analysis in R" class="md-nav__link">
      10X single-cell RNA-seq analysis in R
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-5" type="checkbox" id="nav-5">
    
    <label class="md-nav__link" for="nav-5">
      Proteomics
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-5">
        Proteomics
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../proteomics_basic/proteomics_basic/" title="Identifying proteins from mass spectrometry data" class="md-nav__link">
      Identifying proteins from mass spectrometry data
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../proteomics_basic/background_data_formats/" title="Data Formats Background" class="md-nav__link">
      Data Formats Background
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../proteomics_basic/background_protein_databases/" title="Protein Databases" class="md-nav__link">
      Protein Databases
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../proteomics_basic/background_search_engines/" title="Search Engines" class="md-nav__link">
      Search Engines
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../proteomics_basic/background_protein_prophet/" title="Protein Prophet" class="md-nav__link">
      Protein Prophet
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-6" type="checkbox" id="nav-6">
    
    <label class="md-nav__link" for="nav-6">
      Statistics and Visualisation
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-6">
        Statistics and Visualisation
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../python_viz/python_viz/" title="Data visualisation with Python" class="md-nav__link">
      Data visualisation with Python
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-7" type="checkbox" id="nav-7">
    
    <label class="md-nav__link" for="nav-7">
      Structural Modelling
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-7">
        Structural Modelling
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../molecular_dynamics_101/molecular_dynamics_101/" title="Molecular Dynamics - Introduction to cluster computing" class="md-nav__link">
      Molecular Dynamics - Introduction to cluster computing
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../molecular_dynamics_201/molecular_dynamics_201/" title="Molecular Dynamics - Building input files, visualising the trajectory" class="md-nav__link">
      Molecular Dynamics - Building input files, visualising the trajectory
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#overview" title="Overview" class="md-nav__link">
    Overview
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#1-get-started" title="1. Get started" class="md-nav__link">
    1. Get started
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#mac-users" title="Mac users" class="md-nav__link">
    Mac users
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#windows-users" title="Windows users" class="md-nav__link">
    Windows users
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#activate-the-conda-environment" title="Activate the conda environment" class="md-nav__link">
    Activate the conda environment
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#create-a-new-working-directory-on-your-remote-computer" title="Create a new working directory on your remote computer." class="md-nav__link">
    Create a new working directory on your remote computer.
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#2-get-data" title="2. Get data" class="md-nav__link">
    2. Get data
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#3-assemble" title="3. Assemble" class="md-nav__link">
    3. Assemble
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#4-check-assembly-output" title="4. Check assembly output" class="md-nav__link">
    4. Check assembly output
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#questions" title="Questions" class="md-nav__link">
    Questions
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#5-trim-and-circularise" title="5. Trim and circularise" class="md-nav__link">
    5. Trim and circularise
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#run-circlator" title="Run Circlator" class="md-nav__link">
    Run Circlator
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#check-the-output" title="Check the output" class="md-nav__link">
    Check the output
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#questions_1" title="Questions" class="md-nav__link">
    Questions
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#6-find-smaller-plasmids" title="6. Find smaller plasmids" class="md-nav__link">
    6. Find smaller plasmids
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#align-illumina-reads-to-the-pacbio-contig" title="Align Illumina reads to the PacBio contig" class="md-nav__link">
    Align Illumina reads to the PacBio contig
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#extract-unmapped-illumina-reads" title="Extract unmapped Illumina reads" class="md-nav__link">
    Extract unmapped Illumina reads
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#assemble-the-unmapped-reads" title="Assemble the unmapped reads" class="md-nav__link">
    Assemble the unmapped reads
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#trim-the-plasmid" title="Trim the plasmid" class="md-nav__link">
    Trim the plasmid
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#collect-contigs" title="Collect contigs" class="md-nav__link">
    Collect contigs
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#questions_2" title="Questions" class="md-nav__link">
    Questions
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#7-correct-the-assembly" title="7. Correct the assembly" class="md-nav__link">
    7. Correct the assembly
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#make-an-alignment-file" title="Make an alignment file" class="md-nav__link">
    Make an alignment file
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#run-pilon" title="Run Pilon" class="md-nav__link">
    Run Pilon
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#questions_3" title="Questions" class="md-nav__link">
    Questions
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#8-comparative-genomics" title="8. Comparative Genomics" class="md-nav__link">
    8. Comparative Genomics
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#assemblies" title="Assemblies" class="md-nav__link">
    Assemblies
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#annotations" title="Annotations" class="md-nav__link">
    Annotations
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#9-summary" title="9. Summary" class="md-nav__link">
    9. Summary
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/melbournebioinformatics/MelBioInf_docs/edit/master/docs/tutorials/pacbio/index.md" title="Edit this page" class="md-icon md-content__icon">&#xE3C9;</a>
                
                
                <h1 id="long-read-assembly-workshop">Long read assembly workshop<a class="headerlink" href="#long-read-assembly-workshop" title="Permanent link">&para;</a></h1>
<p>This is a tutorial for a workshop on long-read (PacBio) genome assembly.</p>
<p>It demonstrates how to use long PacBio sequencing reads to assemble a bacterial genome, and includes additional steps for circularising, trimming, finding plasmids, and correcting the assembly with short-read Illumina data.</p>
<h2 id="overview">Overview<a class="headerlink" href="#overview" title="Permanent link">&para;</a></h2>
<p>Simplified version of workflow:</p>
<p><img alt="workflow" src="images/flowchart.png" /></p>
<h2 id="1-get-started">1. Get started<a class="headerlink" href="#1-get-started" title="Permanent link">&para;</a></h2>
<p>Your workshop trainers will provide you with the address of a virtual machine.</p>
<h3 id="mac-users">Mac users<a class="headerlink" href="#mac-users" title="Permanent link">&para;</a></h3>
<p>Open the Terminal.</p>
<ul>
<li>Type in</li>
</ul>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="n">ssh</span> <span class="n">researcher</span><span class="o">@</span><span class="p">[</span><span class="n">your</span> <span class="n">virtual</span> <span class="n">machine</span> <span class="n">address</span><span class="p">]</span>
</pre></div>
</td></tr></table>

<ul>
<li>Type in the password provided.</li>
</ul>
<h3 id="windows-users">Windows users<a class="headerlink" href="#windows-users" title="Permanent link">&para;</a></h3>
<p>If you are using Windows 10, you might be able to use the Ubuntu Subsystem. Otherwise, install and open Putty.</p>
<ul>
<li>Download putty <a href="http://www.chiark.greenend.org.uk/~sgtatham/putty/download.html">here</a>.</li>
<li>Open. A configuration window will appear.</li>
<li>Under &ldquo;Host Name (or IP address)&rdquo; enter in the address of your virtual machine.</li>
<li>Under &ldquo;Port&rdquo; type in 22</li>
<li>Under &ldquo;Connection Type&rdquo; select &ldquo;SSH&rdquo;</li>
<li>Click &ldquo;Open&rdquo;</li>
<li>Under &ldquo;Login as:&rdquo; enter &ldquo;researcher&rdquo;</li>
<li>Type in the password provided.</li>
</ul>
<h3 id="activate-the-conda-environment">Activate the conda environment<a class="headerlink" href="#activate-the-conda-environment" title="Permanent link">&para;</a></h3>
<p>Type in:
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="k">source</span> <span class="o">/</span><span class="n">mnt</span><span class="o">/</span><span class="n">gvl</span><span class="o">/</span><span class="n">apps</span><span class="o">/</span><span class="n">conda</span><span class="o">/</span><span class="n">bin</span><span class="o">/</span><span class="n">activate</span>
</pre></div>
</td></tr></table></p>
<p>(This points us to some different directories for the software we need).</p>
<h3 id="create-a-new-working-directory-on-your-remote-computer">Create a new working directory on your remote computer.<a class="headerlink" href="#create-a-new-working-directory-on-your-remote-computer" title="Permanent link">&para;</a></h3>
<p>Because we are starting a new analysis it is always good practice to start in a new empty directory. Therefore, we will create a new directory and change to it for the rest of the workshop.</p>
<p>In your terminal:</p>
<ul>
<li>Create a new directory called &ldquo;Workshop&rdquo;</li>
</ul>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>mkdir Workshop
</pre></div>
</td></tr></table>

<ul>
<li>Change to that directory</li>
</ul>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>cd Workshop
</pre></div>
</td></tr></table>

<p><strong>NOTE: Every time you open a new terminal or Putty session, you will need to make sure you are in this directory again.</strong></p>
<p>Find your current directory by typing:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>pwd
</pre></div>
</td></tr></table>

<h2 id="2-get-data">2. Get data<a class="headerlink" href="#2-get-data" title="Permanent link">&para;</a></h2>
<p>The sample used in this tutorial is from a bacteria called <em>Staphylococcus aureus</em>. We have used a small section of its real genome so that the programs can run in the workshop time.</p>
<p>The files we need are:</p>
<ul>
<li><fn>pacbio.fq</fn>: the PacBio reads</li>
<li><fn>R1.fq</fn>: the Illumina forward reads</li>
<li><fn>R2.fq</fn>: the Illumina reverse reads</li>
</ul>
<p>In a new tab, go to <a href="https://doi.org/10.5281/zenodo.1009308">https://doi.org/10.5281/zenodo.1009308</a>.</p>
<ul>
<li>Next to the first file, right-click (or control-click) the &ldquo;Download&rdquo; button, and select &ldquo;Copy link address&rdquo;.</li>
<li>Back in your terminal, enter
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>wget [paste file link here]
</pre></div>
</td></tr></table></li>
<li>The file should download.</li>
<li><em>Note:</em> paste the link to the file, not to the webpage.</li>
<li>Repeat this for the other two files.</li>
</ul>
<p>Shorten each of these files names with the <code class="codehilite"><span class="n">mv</span></code> command:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span>1
2
3</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="n">mv</span> <span class="n">R1</span><span class="p">.</span><span class="n">fq</span><span class="err">\</span><span class="o">?</span><span class="n">download</span><span class="err">\</span><span class="o">=</span><span class="mi">1</span> <span class="n">R1</span><span class="p">.</span><span class="n">fq</span>
<span class="n">mv</span> <span class="n">R2</span><span class="p">.</span><span class="n">fq</span><span class="err">\</span><span class="o">?</span><span class="n">download</span><span class="err">\</span><span class="o">=</span><span class="mi">1</span> <span class="n">R2</span><span class="p">.</span><span class="n">fq</span>
<span class="n">mv</span> <span class="n">pacbio</span><span class="p">.</span><span class="n">fq</span><span class="err">\</span><span class="o">?</span><span class="n">download</span><span class="err">\</span><span class="o">=</span><span class="mi">1</span> <span class="n">pacbio</span><span class="p">.</span><span class="n">fq</span>
</pre></div>
</td></tr></table>

<p>Type in <code class="codehilite"><span class="n">ls</span></code> to check the files are present and correctly-named.</p>
<p>We should have <fn>R1.fq</fn>, <fn>R2.fq</fn> and <fn>pacbio.fq</fn>.</p>
<h2 id="3-assemble">3. Assemble<a name="assemble"></a><a class="headerlink" href="#3-assemble" title="Permanent link">&para;</a></h2>
<p>We will use the assembly software called <a href="http://canu.readthedocs.io/en/stable/">Canu</a>, version 1.7.</p>
<p>Run Canu with these commands:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>canu -p canu -d canu_outdir genomeSize=0.03m corThreads=3 -pacbio-raw pacbio.fq
</pre></div>
</td></tr></table>

<ul>
<li>the first <code class="codehilite"><span class="n">canu</span></code> tells the program to run</li>
<li><code class="codehilite"><span class="o">-</span><span class="n">p</span> <span class="n">canu</span></code> names prefix for output files (&ldquo;canu&rdquo;)</li>
<li><code class="codehilite"><span class="o">-</span><span class="n">d</span> <span class="n">canu_outdir</span></code> names output directory (&ldquo;canu_outdir&rdquo;)</li>
<li><code class="codehilite"><span class="n">genomeSize</span></code> only has to be approximate. (In this case we are using a partial genome of expected size 30,000 base pairs).</li>
<li><code class="codehilite"><span class="n">corThreads</span><span class="o">=</span><span class="mi">3</span></code> sets the number of available threads.</li>
<li>Canu will correct, trim and assemble the reads.</li>
<li>Various output will be displayed on the screen.</li>
<li><em>Note</em>: Canu could say &ldquo;Finished&rdquo; but may still be running. In this case, type <code class="codehilite"><span class="n">squeue</span></code> to see if jobs are still running.</li>
</ul>
<p>If you run <code class="codehilite"><span class="n">squeue</span></code> you will see something like this:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span>1
2
3</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
                 6      main canu_can research PD       0:00      1 (Dependency)
               5_1      main cormhap_ research  R       0:29      1 master
</pre></div>
</td></tr></table>

<p>You will know if <strong>Canu</strong> has completely finished when <code class="codehilite"><span class="n">squeue</span></code> shows no jobs listed under the header row.</p>
<h2 id="4-check-assembly-output">4. Check assembly output<a class="headerlink" href="#4-check-assembly-output" title="Permanent link">&para;</a></h2>
<p>Move into the canu output folder:
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="n">cd</span> <span class="n">canu_outdir</span>
</pre></div>
</td></tr></table></p>
<p>View the list of files:
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="n">ls</span>
</pre></div>
</td></tr></table></p>
<ul>
<li>The <fn>canu.contigs.fasta</fn> are the assembled sequences.</li>
<li>The <fn>canu.unassembled.fasta</fn> are the reads that could not be assembled.</li>
<li>The <fn>canu.correctedReads.fasta.gz</fn> are the corrected Pacbio reads that were used in the assembly.</li>
<li>The <fn>canu.contigs.gfa</fn> is the graph of the assembly.</li>
<li>The <fn>canu.report</fn> file is a summary of all of the steps Canu performed with information about the reads used, how they were handled and a whole lot of summary information about the assembly.</li>
</ul>
<p>Display summary information about the contigs: (<code class="codehilite"><span class="n">infoseq</span></code> is a tool from <a href="http://emboss.sourceforge.net/index.html">EMBOSS</a>)</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>infoseq canu.contigs.fasta
</pre></div>
</td></tr></table>

<ul>
<li>This will show the contigs found by Canu. e.g.,  tig00000001  47997</li>
<li>&ldquo;tig00000001&rdquo; is the name given to the contig</li>
<li>&ldquo;47997&rdquo; is the number of base pairs in that contig.</li>
</ul>
<p>This matches what we were expecting for this sample (approximately 30,000 base pairs). For other data, Canu may not be able to join all the reads into one contig, so there may be several contigs in the output.</p>
<p>We should also look at the <fn>canu.report</fn>. To do this:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="k">less</span> <span class="n">canu</span><span class="p">.</span><span class="n">report</span>
</pre></div>
</td></tr></table>

<ul>
<li>&ldquo;less&rdquo; is a command to display the file on the screen.</li>
<li>Use the up and down arrows to scroll up and down.</li>
<li>You will see lots of histograms of read lengths before and after processing, final contig construction, etc.</li>
<li>For a description of the outputs that Canu produces, see: <a href="http://canu.readthedocs.io/en/latest/tutorial.html#outputs">http://canu.readthedocs.io/en/latest/tutorial.html#outputs</a></li>
<li>Type <code class="codehilite"><span class="n">q</span></code> to exit viewing the report.</li>
</ul>
<h3 id="questions">Questions<a class="headerlink" href="#questions" title="Permanent link">&para;</a></h3>
<p>How do long- and short-read assembly methods differ?</p>
<details>
<summary>Answer (click to reveal)</summary>
Short reads are usually assembled with De Bruijn graphs. For long reads, there is a move back towards simpler overlap-layout-consensus methods.
</details>

<p>Where can we find out the what the approximate genome size should be for the species being assembled?</p>
<details>
<summary>Answer (click to reveal)</summary>
Go to NCBI Genomes, enter species name, click on Genome Assembly and Annotation report, sort table by clicking on the column header Size (Mb), look at range of sizes in this column.
</details>

<p>In the assembly output, what are the unassembled reads?</p>
<details>
<summary>Answer (click to reveal)</summary>
Reads and low-coverage contigs that were not used in the assembly.
</details>

<p>What are the corrected reads? How did canu correct the reads?</p>
<details>
<summary>Answer (click to reveal)</summary>
Canu builds overlaps between reads. The consensus is used to correct the reads.
</details>

<p>Where could you view the output .gfa and what would it show?</p>
<details>
<summary>Answer (click to reveal)</summary>
A useful program is [Bandage](https://rrwick.github.io/Bandage/). If the assembly has multiple contigs, the assembly graph shows how these are connected.
</details>

<h2 id="5-trim-and-circularise">5. Trim and circularise<a class="headerlink" href="#5-trim-and-circularise" title="Permanent link">&para;</a></h2>
<p>Bacteria have circular chromosomes.</p>
<ul>
<li>Because of sequencing errors, there may be some &ldquo;overhang&rdquo; in the assembled linear sequence.</li>
<li>Our assembly may have some overhang because it is 9000 bases longer than expected.</li>
</ul>
<p><img alt="circlator" src="images/circlator_diagram.png" />
<em>Adapted from Figure 1. Hunt et al. Genome Biology 2015</em></p>
<p>A tool called <a href="http://sanger-pathogens.github.io/circlator/">Circlator</a> identifies and trims overhangs (on chromosomes and plasmids). It takes in the assembled contigs from Canu, as well as the corrected reads prepared by Canu.</p>
<p>Move back into your main analysis folder:
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="n">cd</span> <span class="p">..</span>
</pre></div>
</td></tr></table></p>
<h3 id="run-circlator">Run Circlator<a class="headerlink" href="#run-circlator" title="Permanent link">&para;</a></h3>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>circlator all --threads 4 --verbose canu_outdir/canu.contigs.fasta canu_outdir/canu.correctedReads.fasta.gz circlator_outdir
</pre></div>
</td></tr></table>

<ul>
<li>
<p>(Click on the dark grey slider bar above and move it to the right, to see all the way to the end of the line.)</p>
</li>
<li>
<p><code class="codehilite"><span class="c1">--threads</span></code> is the number of cores</p>
</li>
<li><code class="codehilite"><span class="c1">--verbose</span></code> prints progress information to the screen</li>
<li><code class="codehilite"><span class="n">canu_outdir</span><span class="o">/</span><span class="n">canu</span><span class="p">.</span><span class="n">contigs</span><span class="p">.</span><span class="n">fasta</span></code> is the file path to the input Canu assembly</li>
<li><code class="codehilite"><span class="n">canu_outdir</span><span class="o">/</span><span class="n">canu</span><span class="p">.</span><span class="n">correctedReads</span><span class="p">.</span><span class="n">fasta</span><span class="p">.</span><span class="n">gz</span></code> is the file path to the corrected Pacbio reads - note, fastA not fastQ</li>
<li><code class="codehilite"><span class="n">circlator_outdir</span></code> is the name of the output directory.</li>
</ul>
<p>Some output will print to screen. When finished, it should say &ldquo;Circularized x of x contig(s)&rdquo;.</p>
<h3 id="check-the-output">Check the output<a class="headerlink" href="#check-the-output" title="Permanent link">&para;</a></h3>
<p>Move into the Circlator output directory:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="n">cd</span> <span class="n">circlator_outdir</span>
</pre></div>
</td></tr></table>

<p>List the files:
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="n">ls</span>
</pre></div>
</td></tr></table></p>
<p>Circlator has named the output files with numbers as prefixes.</p>
<p>Were the contigs circularised?</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="k">less</span> <span class="mi">04</span><span class="p">.</span><span class="n">merge</span><span class="p">.</span><span class="n">circularise</span><span class="p">.</span><span class="n">log</span>
</pre></div>
</td></tr></table>

<ul>
<li>&ldquo;less&rdquo; is a command to display the file on the screen.</li>
<li><fn>04.merge.circularise.log</fn> is the name of the file.</li>
<li>Yes, the contig was circularised (last column).</li>
<li>Type <code class="codehilite"><span class="n">q</span></code> to exit.</li>
</ul>
<p>What are the trimmed contig sizes?
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="n">infoseq</span> <span class="mi">06</span><span class="p">.</span><span class="n">fixstart</span><span class="p">.</span><span class="n">fasta</span>
</pre></div>
</td></tr></table></p>
<ul>
<li>The contig &ldquo;tig00000001&rdquo; has a length of 30019.</li>
<li>This is about 18,000 bases shorter than before circularisation. This was the &ldquo;overhang&rdquo; and has now been trimmed.</li>
</ul>
<p>Copy the circularised contigs file to the main analysis directory with a new name:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="n">cp</span> <span class="mi">06</span><span class="p">.</span><span class="n">fixstart</span><span class="p">.</span><span class="n">fasta</span> <span class="p">..</span><span class="o">/</span><span class="n">contig1</span><span class="p">.</span><span class="n">fasta</span>
</pre></div>
</td></tr></table>

<p>Move back into the main folder:
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="n">cd</span> <span class="p">..</span>
</pre></div>
</td></tr></table></p>
<h3 id="questions_1">Questions<a class="headerlink" href="#questions_1" title="Permanent link">&para;</a></h3>
<p>Were all the contigs circularised?</p>
<details>
<summary>Answer (click to reveal)</summary>
In this example, yes, the contig was circularised.
</details>

<p>Circlator can set the start of the sequence at a particular gene. Which gene does it use? Is this appropriate for all contigs?</p>
<details>
<summary>Answer (click to reveal)</summary>
Circlator uses dnaA (if present) for the chromosomal contig. For other contigs, it uses a centrally-located gene. However, ideally, plasmids would be oriented on a gene such as repA. It is possible to provide a file to Circlator to do this.
</details>

<h2 id="6-find-smaller-plasmids">6. Find smaller plasmids<a class="headerlink" href="#6-find-smaller-plasmids" title="Permanent link">&para;</a></h2>
<p>Pacbio reads are long, and may have been longer than small plasmids. We will look for any small plasmids using the Illumina reads.</p>
<p>This section involves several steps:</p>
<ol>
<li>Use the Canu+Circlator output of a trimmed assembly contig.</li>
<li>Map all the Illumina reads against this Pacbio-assembled contig.</li>
<li>Extract any reads that <em>didn&rsquo;t</em> map and assemble them together: this could be a plasmid, or part of a plasmid.</li>
<li>Look for overhang: if found, trim.</li>
</ol>
<h3 id="align-illumina-reads-to-the-pacbio-contig">Align Illumina reads to the PacBio contig<a class="headerlink" href="#align-illumina-reads-to-the-pacbio-contig" title="Permanent link">&para;</a></h3>
<p>Index the contigs file:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>bwa index contig1.fasta
</pre></div>
</td></tr></table>

<p>Align Illumina reads using using bwa mem:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>bwa mem -t 4 contig1.fasta R1.fq R2.fq | samtools sort &gt; aln.bam
</pre></div>
</td></tr></table>

<ul>
<li><code class="codehilite"><span class="n">bwa</span> <span class="n">mem</span></code> is the alignment tool</li>
<li><code class="codehilite"><span class="o">-</span><span class="n">t</span> <span class="mi">4</span></code> is the number of cores</li>
<li><code class="codehilite"><span class="n">contig1</span><span class="p">.</span><span class="n">fasta</span></code> is the input assembly file</li>
<li><code class="codehilite"><span class="n">R1</span><span class="p">.</span><span class="n">fq</span> <span class="n">R2</span><span class="p">.</span><span class="n">fq</span></code> are the Illumina reads</li>
<li><code class="codehilite"><span class="o">|</span> <span class="n">samtools</span> <span class="n">sort</span></code> pipes the output to samtools to sort</li>
<li><code class="codehilite"><span class="o">&gt;</span> <span class="n">aln</span><span class="p">.</span><span class="n">bam</span></code> sends the alignment to the file <fn>aln.bam</fn></li>
</ul>
<h3 id="extract-unmapped-illumina-reads">Extract unmapped Illumina reads<a class="headerlink" href="#extract-unmapped-illumina-reads" title="Permanent link">&para;</a></h3>
<p>Index the alignment file:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>samtools index aln.bam
</pre></div>
</td></tr></table>

<p>Extract the fastq files from the bam alignment - those reads that were unmapped to the Pacbio alignment - and save them in various &ldquo;unmapped&rdquo; files:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>samtools fastq -f 4 -1 unmapped.R1.fastq -2 unmapped.R2.fastq -s unmapped.RS.fastq aln.bam
</pre></div>
</td></tr></table>

<ul>
<li><code class="codehilite"><span class="n">fastq</span></code> is a command that coverts a <fn>.bam</fn> file into fastq format</li>
<li><code class="codehilite"><span class="o">-</span><span class="n">f</span> <span class="mi">4</span></code> : only output unmapped reads</li>
<li><code class="codehilite"><span class="o">-</span><span class="mi">1</span></code> : put R1 reads into a file called <fn>unmapped.R1.fastq</fn></li>
<li><code class="codehilite"><span class="o">-</span><span class="mi">2</span></code> : put R2 reads into a file called <fn>unmapped.R2.fastq</fn></li>
<li><code class="codehilite"><span class="o">-</span><span class="n">s</span></code> : put singleton reads into a file called <fn>unmapped.RS.fastq</fn></li>
<li><code class="codehilite"><span class="n">aln</span><span class="p">.</span><span class="n">bam</span></code> : input alignment file</li>
</ul>
<p>We now have three files of the unampped reads: <fn> unmapped.R1.fastq</fn>, <fn> unmapped.R2.fastq</fn>, <fn> unmapped.RS.fastq</fn>.</p>
<h3 id="assemble-the-unmapped-reads">Assemble the unmapped reads<a class="headerlink" href="#assemble-the-unmapped-reads" title="Permanent link">&para;</a></h3>
<p>Assemble with Spades:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>spades.py -1 unmapped.R1.fastq -2 unmapped.R2.fastq -s unmapped.RS.fastq --careful --cov-cutoff auto -o spades_assembly
</pre></div>
</td></tr></table>

<ul>
<li>
<p>(Click on the dark grey slider bar above and move it to the right, to see all the way to the end of the line.)</p>
</li>
<li>
<p><code class="codehilite"><span class="o">-</span><span class="mi">1</span></code> is input file forward</p>
</li>
<li><code class="codehilite"><span class="o">-</span><span class="mi">2</span></code> is input file reverse</li>
<li><code class="codehilite"><span class="o">-</span><span class="n">s</span></code> is unpaired</li>
<li><code class="codehilite"><span class="c1">--careful</span></code> minimizes mismatches and short indels</li>
<li><code class="codehilite"><span class="c1">--cov-cutoff auto</span></code> computes the coverage threshold (rather than the default setting, &ldquo;off&rdquo;)</li>
<li><code class="codehilite"><span class="o">-</span><span class="n">o</span></code> is the output directory</li>
</ul>
<p>Move into the output directory:
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="n">cd</span> <span class="n">spades_assembly</span>
</pre></div>
</td></tr></table></p>
<p>Look at the contigs:
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="n">infoseq</span> <span class="n">contigs</span><span class="p">.</span><span class="n">fasta</span>
</pre></div>
</td></tr></table></p>
<ul>
<li>1 contig has been assembled with a length of 2359 bases.</li>
</ul>
<p>Copy it to a new file:
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="n">cp</span> <span class="n">contigs</span><span class="p">.</span><span class="n">fasta</span> <span class="n">contig2</span><span class="p">.</span><span class="n">fasta</span>
</pre></div>
</td></tr></table></p>
<h3 id="trim-the-plasmid">Trim the plasmid<a class="headerlink" href="#trim-the-plasmid" title="Permanent link">&para;</a></h3>
<p>To trim any overhang on this plasmid, we will blast the start of contig2 against itself.</p>
<p>Take the start of the contig:</p>
<p><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="n">head</span> <span class="o">-</span><span class="n">n</span> <span class="mi">10</span> <span class="n">contig2</span><span class="p">.</span><span class="n">fasta</span> <span class="o">&gt;</span> <span class="n">contig2</span><span class="p">.</span><span class="n">fa</span><span class="p">.</span><span class="n">head</span>
</pre></div>
</td></tr></table>
- <code class="codehilite"><span class="n">head</span> <span class="o">-</span><span class="n">n</span> <span class="mi">10</span></code> takes the first ten lines of <fn>contig2.fasta</fn>
- <code class="codehilite"><span class="o">&gt;</span></code> sends that output to a new file called <fn>contig2.fa.head</fn></p>
<ul>
<li>We want to see if the start of the contig matches the end (overhang).</li>
</ul>
<p>Format the assembly file for blast:
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="n">makeblastdb</span> <span class="o">-</span><span class="k">in</span> <span class="n">contig2</span><span class="p">.</span><span class="n">fasta</span> <span class="o">-</span><span class="n">dbtype</span> <span class="n">nucl</span>
</pre></div>
</td></tr></table>
- <code class="codehilite"><span class="n">makeblastdb</span></code> makes a database for the tool Blast
- This will generate three new files in the directory with suffixes .nhr, .nin and .nsq
- <code class="codehilite"><span class="o">-</span><span class="k">in</span></code> sets the input file as <fn> contig2.fasta</fn>
- <code class="codehilite"><span class="o">-</span><span class="n">dbtype</span> <span class="n">nucl</span></code> sets the type to nucleotide (rather than protein)</p>
<p>Blast the start of the assembly (.head file) against all of the assembly:
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="n">blastn</span> <span class="o">-</span><span class="n">query</span> <span class="n">contig2</span><span class="p">.</span><span class="n">fa</span><span class="p">.</span><span class="n">head</span> <span class="o">-</span><span class="n">db</span> <span class="n">contig2</span><span class="p">.</span><span class="n">fasta</span> <span class="o">-</span><span class="n">evalue</span> <span class="mi">1</span><span class="n">e</span><span class="o">-</span><span class="mi">3</span> <span class="o">-</span><span class="n">dust</span> <span class="k">no</span> <span class="o">-</span><span class="k">out</span> <span class="n">contig2</span><span class="p">.</span><span class="n">bls</span>
</pre></div>
</td></tr></table></p>
<ul>
<li><code class="codehilite"><span class="n">blastn</span></code> is the tool Blast, set as blast**n** to compare sequences of nucleotides to each other</li>
<li><code class="codehilite"><span class="o">-</span><span class="n">query</span></code> sets the input sequence as <fn>contig2.fa.head</fn></li>
<li><code class="codehilite"><span class="o">-</span><span class="n">db</span></code> sets the database as that of the original sequence <fn>contig2.fasta</fn>. We don&rsquo;t have to specify the other files that were created when we formatted this file, but they need to present in our current directory.</li>
<li><code class="codehilite"><span class="o">-</span><span class="n">evalue</span></code> is the number of hits expected by chance, here set as 1e-3</li>
<li><code class="codehilite"><span class="o">-</span><span class="n">dust</span> <span class="k">no</span></code> turns off the masking of low-complexity regions</li>
<li><code class="codehilite"><span class="o">-</span><span class="k">out</span></code> sets the output file as <fn>contig2.bls</fn></li>
</ul>
<p>Look at the hits (the matches):
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="k">less</span> <span class="n">contig2</span><span class="p">.</span><span class="n">bls</span>
</pre></div>
</td></tr></table></p>
<ul>
<li>The first hit is at the start, as expected. We can see that &ldquo;Query 1&rdquo; (the start of the contig) is aligned to &ldquo;Sbject 1&rdquo; (the whole contig), for the first 540 bases.</li>
<li>Scroll down with the down arrow.</li>
<li>The second hit shows &ldquo;Query 1&rdquo; (the start of the contig) also matches to &ldquo;Sbject 1&rdquo; (the whole contig) at position 2253, all the way to the end, position 2359.</li>
</ul>
<p><img alt="blast" src="images/blasthits.png" /></p>
<ul>
<li>This is the overhang.</li>
<li>Therefore, in the next step, we need to trim the contig to position 2252.</li>
<li>Type <code class="codehilite"><span class="n">q</span></code> to exit.</li>
</ul>
<p>First, change the name of the contig within the file:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="n">nano</span> <span class="n">contig2</span><span class="p">.</span><span class="n">fasta</span>
</pre></div>
</td></tr></table>

<ul>
<li><code class="codehilite"><span class="n">nano</span></code> opens up a text editor.</li>
<li>Use the arrow keys to navigate. (The mouse won&rsquo;t work.)</li>
<li>At the first line, delete the text, which will be something like &ldquo;&gt;NODE_1_length_2359_cov_3.320333&rdquo;</li>
<li>Type in &ldquo;&gt;contig2&rdquo;</li>
<li>Don&rsquo;t forget the <code class="codehilite"><span class="o">&gt;</span></code> symbol</li>
<li>Press Control-X</li>
<li>&ldquo;Save modified buffer ?&rdquo; - type <code class="codehilite"><span class="n">Y</span></code></li>
<li>Press the Enter key</li>
</ul>
<p>Index the file (this will allow samtools to edit the file as it will have an index):</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="n">samtools</span> <span class="n">faidx</span> <span class="n">contig2</span><span class="p">.</span><span class="n">fasta</span>
</pre></div>
</td></tr></table>

<ul>
<li><code class="codehilite"><span class="n">faidx</span></code> means index the fasta file</li>
</ul>
<p>Trim the contig:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="n">samtools</span> <span class="n">faidx</span> <span class="n">contig2</span><span class="p">.</span><span class="n">fasta</span> <span class="n">contig2</span><span class="p">:</span><span class="mi">1</span><span class="o">-</span><span class="mi">2252</span> <span class="o">&gt;</span> <span class="n">plasmid</span><span class="p">.</span><span class="n">fasta</span>
</pre></div>
</td></tr></table>

<ul>
<li>this extracts contig2 from position 1-2252</li>
<li>
<p><code class="codehilite"><span class="o">&gt;</span> <span class="n">plasmid</span><span class="p">.</span><span class="n">fasta</span></code> sends the extracted section to a new file</p>
</li>
<li>
<p>We now have a trimmed plasmid.</p>
</li>
</ul>
<p>Copy the plasmid file into the main folder:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="n">cp</span> <span class="n">plasmid</span><span class="p">.</span><span class="n">fasta</span> <span class="p">..</span><span class="o">/</span>
</pre></div>
</td></tr></table>

<p>Move file back into main folder:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="n">cd</span> <span class="p">..</span>
</pre></div>
</td></tr></table>

<h3 id="collect-contigs">Collect contigs<a class="headerlink" href="#collect-contigs" title="Permanent link">&para;</a></h3>
<p>Collect the chromosome and the plasmid in one fasta file (they will be 2 records in the file):</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>cat contig1.fasta plasmid.fasta &gt; genome.fasta
</pre></div>
</td></tr></table>

<p>See the contigs and sizes:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>infoseq genome.fasta
</pre></div>
</td></tr></table>

<ul>
<li>chromosome: 30019</li>
<li>plasmid: 2252</li>
</ul>
<h3 id="questions_2">Questions<a class="headerlink" href="#questions_2" title="Permanent link">&para;</a></h3>
<p>Why is this section so complicated?</p>
<details>
<summary>Answer (click to reveal)</summary>
Finding small plasmids is difficult for many reasons! This paper has a nice summary: On the (im)possibility to reconstruct plasmids from whole genome short-read sequencing data. doi: https://doi.org/10.1101/086744
</details>

<p>Why can PacBio sequencing miss small plasmids?</p>
<details>
<summary>Answer (click to reveal)</summary>
Library prep size selection.
</details>

<p>We extract unmapped Illumina reads and assemble these to find small plasmids. What could they be missing?</p>
<details>
<summary>Answer (click to reveal)</summary>
Repeats that have mapped to the PacBio assembly.
</details>

<p>How do you find a plasmid in a Bandage graph?</p>
<details>
<summary>Answer (click to reveal)</summary>
It is probably circular, matches the size of a known plasmid, and has a rep gene.
</details>

<p>Are there easier ways to find plasmids?</p>
<details>
<summary>Answer (click to reveal)</summary>
Possibly. One option is the program called Unicycler which may automate many of these steps. https://github.com/rrwick/Unicycler
</details>

<h2 id="7-correct-the-assembly">7. Correct the assembly<a class="headerlink" href="#7-correct-the-assembly" title="Permanent link">&para;</a></h2>
<p>Sequences from PacBio can have more errors than those from Illumina. Therefore, although it is useful to use the long PacBio reads to assemble the genome, we can also use the shorter and more accurate Illumina reads to correct errors in the PacBio assembly.</p>
<h3 id="make-an-alignment-file">Make an alignment file<a class="headerlink" href="#make-an-alignment-file" title="Permanent link">&para;</a></h3>
<p>Index the fasta file:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>bwa index genome.fasta
</pre></div>
</td></tr></table>

<p>Align the Illumina reads:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>bwa mem -t 4 genome.fasta R1.fq R2.fq | samtools sort &gt; pilon_aln.bam
</pre></div>
</td></tr></table>

<ul>
<li>Aligns Illumina <fn>R1.fq</fn> and <fn>R2.fq</fn> to the PacBio assembly <fn>genome.fasta</fn>.</li>
<li>This produces a .bam file</li>
<li><code class="codehilite"><span class="o">|</span></code> pipes the output to samtools to sort (required for downstream processing)</li>
<li><code class="codehilite"><span class="o">&gt;</span> <span class="n">pilon_aln</span><span class="p">.</span><span class="n">bam</span></code> redirects the sorted bam to this file</li>
</ul>
<p>Index the files:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>samtools index pilon_aln.bam
</pre></div>
</td></tr></table>

<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>samtools faidx genome.fasta
</pre></div>
</td></tr></table>

<ul>
<li>Now we have an alignment file to use  with the tool <a href="https://github.com/broadinstitute/pilon/wiki">Pilon</a>: <fn>pilon_aln.bam</fn></li>
</ul>
<h3 id="run-pilon">Run Pilon<a class="headerlink" href="#run-pilon" title="Permanent link">&para;</a></h3>
<p>Run:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>pilon --genome genome.fasta --frags pilon_aln.bam --output pilon1 --fix all --mindepth 0.5 --changes --verbose --threads 4
</pre></div>
</td></tr></table>

<ul>
<li><code class="codehilite"><span class="c1">--genome</span></code> is the name of the input assembly to be corrected</li>
<li><code class="codehilite"><span class="c1">--frags</span></code> is the alignment of the reads against the assembly</li>
<li><code class="codehilite"><span class="c1">--output</span></code> is the name of the output prefix</li>
<li><code class="codehilite"><span class="c1">--fix</span></code> is an option for types of corrections</li>
<li><code class="codehilite"><span class="c1">--mindepth</span></code> gives a minimum read depth to use</li>
<li><code class="codehilite"><span class="c1">--changes</span></code> produces an output file of the changes made</li>
<li><code class="codehilite"><span class="c1">--verbose</span></code> prints information to the screen during the run</li>
<li><code class="codehilite"><span class="c1">--threads</span></code>: number of cores</li>
</ul>
<p>Look at the changes file:
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="k">less</span> <span class="n">pilon1</span><span class="p">.</span><span class="n">changes</span>
</pre></div>
</td></tr></table></p>
<p><em>Example:</em></p>
<p><img alt="pilon" src="images/pilon.png" /></p>
<ul>
<li>We can see lots of cases where a deletion (represented by a dot) has been corrected to a base.  </li>
<li>Type <code class="codehilite"><span class="n">q</span></code> to exit.</li>
</ul>
<p>Look at the details of the fasta file:
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="n">infoseq</span> <span class="n">pilon1</span><span class="p">.</span><span class="n">fasta</span>
</pre></div>
</td></tr></table></p>
<ul>
<li>chromosome - 30059 (net +40 bases)</li>
<li>plasmid - 2252 (no change)</li>
</ul>
<p>Change the file name:
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="n">cp</span> <span class="n">pilon1</span><span class="p">.</span><span class="n">fasta</span> <span class="n">assembly</span><span class="p">.</span><span class="n">fasta</span>
</pre></div>
</td></tr></table></p>
<p>We now have the corrected genome assembly of <em>Staphylococcus aureus</em> in .fasta format, containing a chromosome and a small plasmid.  </p>
<h3 id="questions_3">Questions<a class="headerlink" href="#questions_3" title="Permanent link">&para;</a></h3>
<p>Why don&rsquo;t we correct earlier in the assembly process?</p>
<details>
<summary>Answer (click to reveal)</summary>
We need to circularise the contigs and trim overhangs first.
</details>

<p>Why can we use some reads (Illumina) to correct other reads (PacBio) ?</p>
<details>
<summary>Answer (click to reveal)</summary>
Illumina reads have higher accuracy.
</details>

<p>Could we just use PacBio reads to assemble the genome?</p>
<details>
<summary>Answer (click to reveal)</summary>
Yes, if accuracy adequate.
</details>

<h2 id="8-comparative-genomics">8. Comparative Genomics<a class="headerlink" href="#8-comparative-genomics" title="Permanent link">&para;</a></h2>
<p>In the workshop so far, we used a partial bacterial genome so that the exercises could run in the time available. As a demonstration, to better see the effect of long and short reads on the assembly, we will examine a complete bacterial genome.</p>
<h3 id="assemblies">Assemblies<a class="headerlink" href="#assemblies" title="Permanent link">&para;</a></h3>
<p>This bacterial genome has been assembled from either long PacBio reads (using Canu) or shorter Illumina reads (using Spades).</p>
<p>Assembly graphs:</p>
<p>Look at the assembly graph (usually has a suffix .gfa), in the program <a href="https://rrwick.github.io/Bandage/">Bandage</a>. This shows how contigs are related, albeit with ambiguity in some places.</p>
<p>The assembly graph from Illumina reads (Spades assembly):</p>
<p><img alt="spades assembly graph" src="images/25747_spades_illumina_graph.png" /></p>
<p>The assembly graph from PacBio reads (Canu assembly) - this is missing the small plasmid:</p>
<p><img alt="canu assembly graph" src="images/25747_canu_pacbio_graph.png" /></p>
<p>Here we can see that the long read data results in a more contiguous assembly - one complete chromosome versus many smaller contigs with ambiguous placement.</p>
<p>Does it matter that an assembly is in many contigs?</p>
<details>
<summary>Answer (click to reveal)</summary>
Yes and No. Yes: broken genes can lead to missing/incorrect annotations; fragmented assemblies provide less information about the genomic structure (*e.g.* the number of plasmids) and the location of genes of interest (*e.g.* gene A is located on plasmid X). No: many or all genes may still be annotated correctly. Gene location is useful (e.g. chromosome, plasmid1) but not always essential (e.g. presence/absence of particular resistance genes may be enough information).
</details>

<h3 id="annotations">Annotations<a class="headerlink" href="#annotations" title="Permanent link">&para;</a></h3>
<p>Genomic features such as genes can be identified with annotation tools. We have used a tool called <a href="https://github.com/tseemann/prokka">Prokka</a> to annotate the two genomes described above.</p>
<p>Some of the output data is displayed here:</p>
<p><center></p>
<table>
<thead>
<tr>
<th>assembly:</th>
<th align="right">PacBio</th>
<th align="right">Illumina</th>
</tr>
</thead>
<tbody>
<tr>
<td>size</td>
<td align="right">2,825,804</td>
<td align="right">2,792,905</td>
</tr>
<tr>
<td>contigs</td>
<td align="right">2</td>
<td align="right">123</td>
</tr>
<tr>
<td>CDS</td>
<td align="right">2614</td>
<td align="right">2575</td>
</tr>
<tr>
<td>tRNA</td>
<td align="right">61</td>
<td align="right">65</td>
</tr>
<tr>
<td>rRNA</td>
<td align="right">19</td>
<td align="right">4</td>
</tr>
</tbody>
</table>
<p></center></p>
<p>Why are there more CDS identified in the PacBio assembly?</p>
<details>
<summary>Answer (click to reveal)</summary>
The PacBio assembly may have errors (usually a one base indel) which will cause a frame shift, which can result in three things: a longer CDS, a shorter CDS, or a shorter CDS plus an additional CDS. In addition, the Illumina assembly is about 33 kb smaller than the PacBio assembly. In bacteria, a rule of thumb is that 1 kb is roughly equal to one gene. Thus, we would probably expect about 33 fewer identified genes, which fits with these results.  
</details>

<p>Why are there more rRNA identified in the PacBio assembly?</p>
<details>
<summary>Answer (click to reveal)</summary>
There may be multiple copies of the rRNAs and these could have been collapsed as repeats in the Illumina assembly.
</details>

<h2 id="9-summary">9. Summary<a class="headerlink" href="#9-summary" title="Permanent link">&para;</a></h2>
<p>In this workshop, we used bacterial sequencing data from long and short reads to produce a polished genome.</p>
<p>Procedure and tools:</p>
<ul>
<li>Canu to assemble long-read PacBio data</li>
<li>Circlator to trim and circularise contigs</li>
<li>BWA-MEM to map shorter Illumina reads to the PacBio assembly</li>
<li>Spades to assemble any unmapped, leftover Illumina reads (the plasmid)</li>
<li>Pilon to correct the PacBio assembly with the more accurate Illumina reads</li>
</ul>
<p>We also looked at comparative genomics:</p>
<ul>
<li>Bandage to examine assembly graphs</li>
<li>Prokka to annotate genomes with features such as genes</li>
</ul>
<p>Further research:</p>
<ul>
<li>Align genomes with Mauve: <a href="http://sepsis-omics.github.io/tutorials/modules/mauve/">tutorial link</a></li>
<li>Find core and pan genomes with Roary and Phandango: <a href="http://sepsis-omics.github.io/tutorials/modules/roary/">tutorial link</a></li>
</ul>
<p>Melbourne Bioinformatics tutorials:</p>
<ul>
<li><a href="https://www.melbournebioinformatics.org.au/tutorials/">https://www.melbournebioinformatics.org.au/tutorials/</a></li>
</ul>
<p>Additional microbial genomics tutorials:</p>
<ul>
<li><a href="http://sepsis-omics.github.io/tutorials/">http://sepsis-omics.github.io/tutorials/</a> and <a href="https://galaxy-au-training.github.io/tutorials/">https://galaxy-au-training.github.io/tutorials/</a></li>
</ul>
                
                  
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../assembly/spades/" title="de novo assembly of Illumina reads using Spades (Galaxy)" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                de novo assembly of Illumina reads using Spades (Galaxy)
              </span>
            </div>
          </a>
        
        
          <a href="../hybrid_assembly/nanopore_assembly/" title="Hybrid genome assembly - Nanopore and Illumina" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                Hybrid genome assembly - Nanopore and Illumina
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            <a href="https://www.melbournebioinformatics.org.au/">Melbourne Bioinformatics</a>
          </div>
        
        powered by
        <a href="https://www.mkdocs.org">MkDocs</a>
        and
        <a href="https://squidfunk.github.io/mkdocs-material/">
          Material for MkDocs</a>
      </div>
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../assets/javascripts/application.245445c6.js"></script>
      
      <script>app.initialize({version:"1.0.4",url:{base:"../.."}})</script>
      
    
  </body>
</html>