



<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
      
      
      
        <meta name="lang:clipboard.copy" content="Copy to clipboard">
      
        <meta name="lang:clipboard.copied" content="Copied to clipboard">
      
        <meta name="lang:search.language" content="en">
      
        <meta name="lang:search.pipeline.stopwords" content="True">
      
        <meta name="lang:search.pipeline.trimmer" content="True">
      
        <meta name="lang:search.result.none" content="No matching documents">
      
        <meta name="lang:search.result.one" content="1 matching document">
      
        <meta name="lang:search.result.other" content="# matching documents">
      
        <meta name="lang:search.tokenizer" content="[\s\-]+">
      
      <link rel="shortcut icon" href="../../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.0.4, mkdocs-material-4.4.0">
    
    
      
        <title>Introduction to Genome Browsers - Bioinformatics Documentation</title>
      
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/application.0284f74d.css">
      
      
    
    
      <script src="../../../assets/javascripts/modernizr.74668098.js"></script>
    
    
      
        <link href="https://fonts.gstatic.com" rel="preconnect" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono&display=fallback">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
    
    <link rel="stylesheet" href="../../../assets/fonts/material-icons.css">
    
    
      <link rel="stylesheet" href="../../../extra.css">
    
    
      
        
<script>
  window.ga = window.ga || function() {
    (ga.q = ga.q || []).push(arguments)
  }
  ga.l = +new Date
  /* Setup integration and send page view */
  ga("create", "UA-21794901-5", "www.melbournebioinformatics.org.au/tutorials/")
  ga("set", "anonymizeIp", true)
  ga("send", "pageview")
  /* Register handler to log search on blur */
  document.addEventListener("DOMContentLoaded", () => {
    if (document.forms.search) {
      var query = document.forms.search.query
      query.addEventListener("blur", function() {
        if (this.value) {
          var path = document.location.pathname;
          ga("send", "pageview", path + "?q=" + this.value)
        }
      })
    }
  })
</script>
<script async src="https://www.google-analytics.com/analytics.js"></script>
      
    
    
  </head>
  
    <body dir="ltr">
  
    <svg class="md-svg">
      <defs>
        
        
          <svg xmlns="http://www.w3.org/2000/svg" width="416" height="448" viewBox="0 0 416 448" id="__github"><path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19T128 352t-18.125-8.5-10.75-19T96 304t3.125-20.5 10.75-19T128 256t18.125 8.5 10.75 19T160 304zm160 0q0 10-3.125 20.5t-10.75 19T288 352t-18.125-8.5-10.75-19T256 304t3.125-20.5 10.75-19T288 256t18.125 8.5 10.75 19T320 304zm40 0q0-30-17.25-51T296 232q-10.25 0-48.75 5.25Q229.5 240 208 240t-39.25-2.75Q130.75 232 120 232q-29.5 0-46.75 21T56 304q0 22 8 38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0 37.25-1.75t35-7.375 30.5-15 20.25-25.75T360 304zm56-44q0 51.75-15.25 82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5T212 416q-19.5 0-35.5-.75t-36.875-3.125-38.125-7.5-34.25-12.875T37 371.5t-21.5-28.75Q0 312 0 260q0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25 30.875Q171.5 96 212 96q37 0 70 8 26.25-20.5 46.75-30.25T376 64q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34 99.5z"/></svg>
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" data-md-component="overlay" for="__drawer"></label>
    
      <a href="#introduction-to-genome-browsers" tabindex="1" class="md-skip">
        Skip to content
      </a>
    
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        <a href="../../.." title="Bioinformatics Documentation" class="md-header-nav__button md-logo">
          
            <i class="md-icon"></i>
          
        </a>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="__drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <div class="md-flex__ellipsis md-header-nav__title" data-md-component="title">
          
            <span class="md-header-nav__topic">
              Bioinformatics Documentation
            </span>
            <span class="md-header-nav__topic">
              
                Introduction to Genome Browsers
              
            </span>
          
        </div>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <label class="md-icon md-icon--search md-header-nav__button" for="__search"></label>
          
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query" data-md-state="active">
      <label class="md-icon md-search__icon" for="__search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset" tabindex="-1">
        &#xE5CD;
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
        
      </div>
      
        <div class="md-flex__cell md-flex__cell--shrink">
          <div class="md-header-nav__source">
            


  

<a href="https://github.com/melbournebioinformatics/MelBioInf_docs/" title="Go to repository" class="md-source" data-md-source="github">
  
    <div class="md-source__icon">
      <svg viewBox="0 0 24 24" width="24" height="24">
        <use xlink:href="#__github" width="24" height="24"></use>
      </svg>
    </div>
  
  <div class="md-source__repository">
    melbournebioinformatics/MelBioInf_docs
  </div>
</a>
          </div>
        </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
        
      
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="__drawer">
    <a href="../../.." title="Bioinformatics Documentation" class="md-nav__button md-logo">
      
        <i class="md-icon"></i>
      
    </a>
    Bioinformatics Documentation
  </label>
  
    <div class="md-nav__source">
      


  

<a href="https://github.com/melbournebioinformatics/MelBioInf_docs/" title="Go to repository" class="md-source" data-md-source="github">
  
    <div class="md-source__icon">
      <svg viewBox="0 0 24 24" width="24" height="24">
        <use xlink:href="#__github" width="24" height="24"></use>
      </svg>
    </div>
  
  <div class="md-source__repository">
    melbournebioinformatics/MelBioInf_docs
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href="../../.." title="Home" class="md-nav__link">
      Home
    </a>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-2" type="checkbox" id="nav-2" checked>
    
    <label class="md-nav__link" for="nav-2">
      Tools and skill development
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-2">
        Tools and skill development
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../../guides/bioinfo/" title="What is Bioinformatics?" class="md-nav__link">
      What is Bioinformatics?
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../guides/galaxy/" title="Introduction to the Galaxy Platform" class="md-nav__link">
      Introduction to the Galaxy Platform
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../galaxy-workflows/galaxy-workflows/" title="Introduction to Galaxy Workflows" class="md-nav__link">
      Introduction to Galaxy Workflows
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../python_overview/python_overview/" title="Introduction to Python" class="md-nav__link">
      Introduction to Python
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../using_git/Using_Git/" title="Introduction to Git and Github" class="md-nav__link">
      Introduction to Git and Github
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../hpc/hpc/" title="Introduction to HPC" class="md-nav__link">
      Introduction to HPC
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../unix/unix/" title="Introduction to Unix" class="md-nav__link">
      Introduction to Unix
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="__toc">
    
      
    
    
      <label class="md-nav__link md-nav__link--active" for="__toc">
        Introduction to Genome Browsers
      </label>
    
    <a href="./" title="Introduction to Genome Browsers" class="md-nav__link md-nav__link--active">
      Introduction to Genome Browsers
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#overview" title="Overview" class="md-nav__link">
    Overview
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#topic" title="Topic" class="md-nav__link">
    Topic
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#skill-level" title="Skill level" class="md-nav__link">
    Skill level
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#description" title="Description" class="md-nav__link">
    Description
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#learning-objectives" title="Learning Objectives" class="md-nav__link">
    Learning Objectives
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#requirements-and-preparation" title="Requirements and preparation" class="md-nav__link">
    Requirements and preparation
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#preparing-your-laptop-prior-to-starting-this-workshop" title="Preparing your laptop prior to starting this workshop" class="md-nav__link">
    Preparing your laptop prior to starting this workshop
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#required-software" title="Required Software" class="md-nav__link">
    Required Software
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#required-data" title="Required Data" class="md-nav__link">
    Required Data
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#authors-and-review-date" title="Author(s) and review date" class="md-nav__link">
    Author(s) and review date
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#background" title="Background" class="md-nav__link">
    Background
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#section-1-introduction-to-genome-browsers" title="Section 1: Introduction to Genome Browsers" class="md-nav__link">
    Section 1: Introduction to Genome Browsers
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#genome-build-version-number" title="Genome Build version number" class="md-nav__link">
    Genome Build version number
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#section-2-ucsc-genome-browser" title="Section 2: UCSC genome Browser" class="md-nav__link">
    Section 2: UCSC genome Browser
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#getting-started" title="Getting started" class="md-nav__link">
    Getting started
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#understanding-the-gene-models" title="Understanding the gene models" class="md-nav__link">
    Understanding the gene models
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gene-expression-data" title="Gene expression data" class="md-nav__link">
    Gene expression data
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#blat-tool" title="Blat tool" class="md-nav__link">
    Blat tool
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#section-3-igv" title="Section 3: IGV" class="md-nav__link">
    Section 3: IGV
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#additional-reading" title="Additional reading" class="md-nav__link">
    Additional reading
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
    
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../rna_seq_exp_design/rna_seq_experimental_design/" title="RNA-seq Experimental Design" class="md-nav__link">
      RNA-seq Experimental Design
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../docker/docker/" title="Containerized Bioinformatics" class="md-nav__link">
      Containerized Bioinformatics
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../cwl/cwl/" title="Common Workflow Language for Bioinformatics" class="md-nav__link">
      Common Workflow Language for Bioinformatics
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../bionitio/bionitio/" title="Command-line software development" class="md-nav__link">
      Command-line software development
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-3" type="checkbox" id="nav-3">
    
    <label class="md-nav__link" for="nav-3">
      Genomics
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-3">
        Genomics
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-3-1" type="checkbox" id="nav-3-1">
    
    <label class="md-nav__link" for="nav-3-1">
      Variant detection
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="2">
      <label class="md-nav__title" for="nav-3-1">
        Variant detection
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../var_detect_advanced/var_detect_advanced_background/" title="Introduction to Variant detection" class="md-nav__link">
      Introduction to Variant detection
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../variant_calling_galaxy_1/variant_calling_galaxy_1/" title="Variant Calling part 1 (Galaxy)" class="md-nav__link">
      Variant Calling part 1 (Galaxy)
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../var_detect_advanced/var_detect_advanced/" title="Variant Calling part 2 (Galaxy)" class="md-nav__link">
      Variant Calling part 2 (Galaxy)
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

        
          
          
          


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-3-2" type="checkbox" id="nav-3-2">
    
    <label class="md-nav__link" for="nav-3-2">
      de novo assembly
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="2">
      <label class="md-nav__title" for="nav-3-2">
        de novo assembly
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../assembly/assembly-background/" title="Introduction to de novo assembly with Velvet" class="md-nav__link">
      Introduction to de novo assembly with Velvet
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../assembly/assembly-protocol/" title="Introduction to de novo genome assembly for Illumina reads" class="md-nav__link">
      Introduction to de novo genome assembly for Illumina reads
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../assembly/assembly/" title="de novo assembly of Illumina reads using Velvet (Galaxy)" class="md-nav__link">
      de novo assembly of Illumina reads using Velvet (Galaxy)
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../assembly/spades/" title="de novo assembly of Illumina reads using Spades (Galaxy)" class="md-nav__link">
      de novo assembly of Illumina reads using Spades (Galaxy)
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../pacbio/" title="de novo assembly of PacBio reads (unix)" class="md-nav__link">
      de novo assembly of PacBio reads (unix)
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../hybrid_assembly/nanopore_assembly/" title="Hybrid genome assembly - Nanopore and Illumina" class="md-nav__link">
      Hybrid genome assembly - Nanopore and Illumina
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-4" type="checkbox" id="nav-4">
    
    <label class="md-nav__link" for="nav-4">
      Transcriptomics
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-4">
        Transcriptomics
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../rna_seq_exp_design/rna_seq_experimental_design/" title="RNAseq Experimental Design" class="md-nav__link">
      RNAseq Experimental Design
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-4-2" type="checkbox" id="nav-4-2">
    
    <label class="md-nav__link" for="nav-4-2">
      Bulk RNAseq
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="2">
      <label class="md-nav__title" for="nav-4-2">
        Bulk RNAseq
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../rna_seq_dge_basic/rna_seq_basic_background/" title="Introduction to bulk RNAseq analysis" class="md-nav__link">
      Introduction to bulk RNAseq analysis
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../rna_seq_dge_basic/rna_seq_basic_tutorial/" title="RNAseq analysis using HISAT2 (Galaxy)" class="md-nav__link">
      RNAseq analysis using HISAT2 (Galaxy)
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../rna_seq_dge_basic/rna_seq_basic_tuxedo/" title="RNAseq analysis using Tuxedo (Galaxy)" class="md-nav__link">
      RNAseq analysis using Tuxedo (Galaxy)
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../rna_seq_dge_advanced/rna_seq_advanced_tutorial/" title="RNAseq differential expression tool comparision (Galaxy)" class="md-nav__link">
      RNAseq differential expression tool comparision (Galaxy)
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../rna_seq_dge_in_r/rna_seq_r/" title="RNAseq differential expression in R" class="md-nav__link">
      RNAseq differential expression in R
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

        
          
          
          


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-4-3" type="checkbox" id="nav-4-3">
    
    <label class="md-nav__link" for="nav-4-3">
      Singlecell RNAseq
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="2">
      <label class="md-nav__title" for="nav-4-3">
        Singlecell RNAseq
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../singlecell/" title="10X single-cell RNA-seq analysis in R" class="md-nav__link">
      10X single-cell RNA-seq analysis in R
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-5" type="checkbox" id="nav-5">
    
    <label class="md-nav__link" for="nav-5">
      Proteomics
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-5">
        Proteomics
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../proteomics_basic/proteomics_basic/" title="Identifying proteins from mass spectrometry data" class="md-nav__link">
      Identifying proteins from mass spectrometry data
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../proteomics_basic/background_data_formats/" title="Data Formats Background" class="md-nav__link">
      Data Formats Background
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../proteomics_basic/background_protein_databases/" title="Protein Databases" class="md-nav__link">
      Protein Databases
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../proteomics_basic/background_search_engines/" title="Search Engines" class="md-nav__link">
      Search Engines
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../proteomics_basic/background_protein_prophet/" title="Protein Prophet" class="md-nav__link">
      Protein Prophet
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-6" type="checkbox" id="nav-6">
    
    <label class="md-nav__link" for="nav-6">
      Statistics and Visualisation
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-6">
        Statistics and Visualisation
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../python_viz/python_viz/" title="Data visualisation with Python" class="md-nav__link">
      Data visualisation with Python
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-7" type="checkbox" id="nav-7">
    
    <label class="md-nav__link" for="nav-7">
      Structural Modelling
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-7">
        Structural Modelling
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../molecular_dynamics_101/molecular_dynamics_101/" title="Molecular Dynamics - Introduction to cluster computing" class="md-nav__link">
      Molecular Dynamics - Introduction to cluster computing
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../molecular_dynamics_201/molecular_dynamics_201/" title="Molecular Dynamics - Building input files, visualising the trajectory" class="md-nav__link">
      Molecular Dynamics - Building input files, visualising the trajectory
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#overview" title="Overview" class="md-nav__link">
    Overview
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#topic" title="Topic" class="md-nav__link">
    Topic
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#skill-level" title="Skill level" class="md-nav__link">
    Skill level
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#description" title="Description" class="md-nav__link">
    Description
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#learning-objectives" title="Learning Objectives" class="md-nav__link">
    Learning Objectives
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#requirements-and-preparation" title="Requirements and preparation" class="md-nav__link">
    Requirements and preparation
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#preparing-your-laptop-prior-to-starting-this-workshop" title="Preparing your laptop prior to starting this workshop" class="md-nav__link">
    Preparing your laptop prior to starting this workshop
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#required-software" title="Required Software" class="md-nav__link">
    Required Software
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#required-data" title="Required Data" class="md-nav__link">
    Required Data
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#authors-and-review-date" title="Author(s) and review date" class="md-nav__link">
    Author(s) and review date
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#background" title="Background" class="md-nav__link">
    Background
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#section-1-introduction-to-genome-browsers" title="Section 1: Introduction to Genome Browsers" class="md-nav__link">
    Section 1: Introduction to Genome Browsers
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#genome-build-version-number" title="Genome Build version number" class="md-nav__link">
    Genome Build version number
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#section-2-ucsc-genome-browser" title="Section 2: UCSC genome Browser" class="md-nav__link">
    Section 2: UCSC genome Browser
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#getting-started" title="Getting started" class="md-nav__link">
    Getting started
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#understanding-the-gene-models" title="Understanding the gene models" class="md-nav__link">
    Understanding the gene models
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gene-expression-data" title="Gene expression data" class="md-nav__link">
    Gene expression data
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#blat-tool" title="Blat tool" class="md-nav__link">
    Blat tool
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#section-3-igv" title="Section 3: IGV" class="md-nav__link">
    Section 3: IGV
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#additional-reading" title="Additional reading" class="md-nav__link">
    Additional reading
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/melbournebioinformatics/MelBioInf_docs/edit/master/docs/tutorials/Genome_browsers/GenomeBrowsers_Intro.md" title="Edit this page" class="md-icon md-content__icon">&#xE3C9;</a>
                
                
                <p><img src="../media/melbioinf_logo.png" width="350"> <img src="../media/PRIMARY_A_Vertical_Housed_RGB.png" width="150"></p>
<h1 id="introduction-to-genome-browsers">Introduction to Genome Browsers</h1>
<p>Anticipated workshop duration when delivered to a group of participants is <strong>3 hours</strong>.<br>
Note that not all exercises are expected to be completed during the workshop.  </p>
<p>For queries relating to this workshop, contact Melbourne Bioinformatics at bioinformatics-training@unimelb.edu.au.</p>
<h2 id="overview">Overview</h2>
<p>This tutorial will introduce you to the genome browser format and illustrate how some freely available genome browsers can be used to interrogate a variety of data types, such as gene expression, genomic variation, methylation and many more.</p>
<h3 id="topic">Topic</h3>
<ul>
<li>[x] Genomics</li>
<li>[x] Transcriptomics</li>
<li>[ ] Proteomics</li>
<li>[ ] Metabolomics</li>
<li>[ ] Statistics and visualisation</li>
<li>[ ] Structural Modelling</li>
<li>[x] Basic skills</li>
</ul>
<h3 id="skill-level">Skill level</h3>
<ul>
<li>[x] Beginner  </li>
<li>[ ] Intermediate  </li>
<li>[ ] Advanced  </li>
</ul>
<p>This workshop is designed for participants with no previous experience of using Genome Browsers and no programming experience.</p>
<h3 id="description">Description</h3>
<p><em>learn how to make the most of Genome Browsers !</em></p>
<p>Genome browsers are invaluable for viewing and interpreting any data that can be anchored to a position on a genome. These include gene location, genomic variation, transcription, and the many types of regulatory data such as methylation sites and transcription factor binding sites. By enabling the viewing of one type of data in the context of other data types the use of genome browsers can reveal important information about genomic variation, and conservation, in gene expression in development and disease.</p>
<p>This tutorial is in three parts.</br></p>
<ul>
<li>Section 1 Introduction to the general features of genome browsers. </br></li>
<li>Section 2 Hands on tutorial of the <a href="https://genome.ucsc.edu/">UCSC Genome Browser</a></li>
<li>Section 3 Hands on tutorials of the <a href="https://software.broadinstitute.org/software/igv/home">Integrative Genomics Viewer</a></li>
</ul>
<p>This tutorial was developed for use as part a series of workshops for neuroscience researchers, hence the data and example genes are drawn from neuroscience field and focused on analysis and visualisation  of expression data. However, the skills taught in this tutorial are applicable to all areas of research.</p>
<p><strong>Data:</strong> <a href="https://gtexportal.org/home/">GTEX</a> and <a href="https://tabula-muris.ds.czbiohub.org/">Tabular Muris</a> data as represented in the UCSC Genome Browser, and <a href="http://casestudies.brain-map.org/celltax">Celltax single cell expression atlas data</a> downloaded from <a href="https://www.ncbi.nlm.nih.gov/sra">SRA</a></br>
<strong>Tools:</strong> <a href="https://genome.ucsc.edu/">UCSC genome Browser</a>, <a href="https://software.broadinstitute.org/software/igv/home">Integrative Genomics Viewer</a></p>
<hr />
<h2 id="learning-objectives">Learning Objectives</h2>
<p>At the end of this introductory workshop, you will :</p>
<ul>
<li>Understand the how different types of genomics and expression data are represented in Genome browsers</li>
<li>Create, save and share custom views data in Genome browsers</li>
<li>Know how to import and view your own data in IGV</li>
<li>Know the main file types used for storing expression data</li>
<li>Know how to index and sort BAM files using IGV tools</li>
</ul>
<hr />
<h2 id="requirements-and-preparation">Requirements and preparation</h2>
<p><strong>Attendees are required to provide their own laptop computers.</strong>  </p>
<p>If delivered as a workshop, participants should install the software and data files below prior to the workshop.  Ensure that you provide sufficient time to liaise with your own IT support should they encounter any IT problems with installing software.  </p>
<p>Unless stated otherwise, recommended browsers are Firefox or Chrome.</p>
<p>Create a user account in UCSC genome browser</p>
<h3 id="preparing-your-laptop-prior-to-starting-this-workshop">Preparing your laptop prior to starting this workshop</h3>
<ol>
<li>Required software:</li>
<li>Ensure that (<a href="https://www.google.com/chrome/">Chrome</a> or <a href="https://www.mozilla.org/en-US/">FireFox</a> are installed and upto date)</br></li>
<li>Download and install <a href="https://software.broadinstitute.org/software/igv/download">IGV</a> (free) on your laptop.</br></li>
<li>Check that the IGV software and data are correctly installed.</li>
<li>Required data is downloaded as part of the tutorial exercises.</li>
</ol>
<h3 id="required-software">Required Software</h3>
<ul>
<li><a href="https://software.broadinstitute.org/software/igv/download">IGV</a> (Free)</li>
</ul>
<h3 id="required-data">Required Data</h3>
<ul>
<li>No additional data needs to be downloaded prior to this workshop.</li>
</ul>
<hr />
<h3 id="authors-and-review-date">Author(s) and review date</h3>
<p>Written by: Victoria Perreau<br />
Melbourne Bioinformatics, University of Melbourne</p>
<p>Created/Reviewed: October 2020</p>
<hr />
<h2 id="background">Background</h2>
<p>Genome browsers are invaluable for viewing and interpreting the many different types of data that can be anchored to genomic positions.  These include variation, transcription, the many types regulatory data such as methylation and transcription factor binding, and disease associations. The larger genome browsers serve as data archives for valuable public datasets facilitating visualisation and analysis of different data types. It is also possible to load your own data into some of the public genome browsers.</p>
<p>By enabling viewing of one type of data in the context of another, the use of Genome browsers can reveal important information about gene regulation in both normal development and disease, assist hypothesis development relating to genotype phenotype relationships.</p>
<p>All researchers are therefore encouraged to become familiar with the use of some of the main browsers such as:</p>
<ul>
<li><a href="https://genome.ucsc.edu/">The UCSC Genome Browser</a>, (RRID:SCR_005780)</li>
<li><a href="https://www.ensembl.org/index.html">ESEMBL Genome Browser</a>, (RRID:SCR_013367)</li>
<li><a href="https://epigenomegateway.wustl.edu/browser/">Epigenome browser at WashU</a>, (RRID:SCR_006208)</li>
<li><a href="http://software.broadinstitute.org/software/igv/">Integrative Genomics Viewer (IGV)</a>, RRID:SCR_011793).</li>
</ul>
<p>They are designed for use by researchers without programming experience and the developers often provide extensive tutorials and cases studies demonstrating the myriad of ways in which data can be loaded and interpreted to assist in develop and supporting your research hypothesis.</p>
<p>Many large genomic projects also incorporate genome browsers into their web portals to enable users to easily search and view the data. These include:</p>
<ul>
<li><a href="https://gtexportal.org/home/">GTEx</a></li>
<li><a href="https://gnomad.broadinstitute.org/">gnomAD</a></li>
</ul>
<hr />
<h2 id="section-1-introduction-to-genome-browsers">Section 1: Introduction to Genome Browsers</h2>
<p>Genome browsers rely on a common reference genome for each species in order to map data from different sources to the correct location. A consortium agreed on a common numbering for each position on the genome,</p>
<p>The sequence for human reference genome was accumulated up over many years from sequence data from many different sources and does not represent the sequence of one single person. Instead it is a composite of fragments of the genome from many different people.  Also, unlike the human genome which is diploid, the human genome is haploid.  That is there is only one copy of each chromosome. It therefore does not reflect the variation on the population, or even the most common variants in the human genome.</p>
<h3 id="genome-build-version-number">Genome Build version number</h3>
<p><a href="https://www.ncbi.nlm.nih.gov/grc">The Genome reference consortium</a></br>
How does the nomenclature mean? https://genome.ucsc.edu/FAQ/FAQreleases.html</p>
<p>For further info on Human Genome version updates I recommend you look at the updates and <a href="https://genome.ucsc.edu/goldenPath/newsarch.html#2019">blog pages on the UCSC genome browser</a>.</p>
<hr />
<h2 id="section-2-ucsc-genome-browser">Section 2: UCSC genome Browser</h2>
<p>In this section we will become familiar with the web interface of the UCSC genome browser and explore some of the tools and public datasets available.</p>
<pre><code>* Explore features of particular chromosomal regions
* Investigate specific genes as well as collections of genes
* Search for locations of sequences and markers
* Retrieve annotation information for specific regions or genome-wide
* View your own data in context of other annotations
* Compare a region of one genome to genomes of other species
</code></pre>
<p>Weekly maintenance of the browser is at 5-6 pm Thursdays Pacific time, which is equivalent to 11am-12pm AEST time. During this time the browser may be down for a few minutes. To ensure uninterrupted browser services for your research during UCSC server maintenance and power outages, bookmark one of the mirror sites that replicates the UCSC genome browser.</p>
<p><strong>Accessing the tools</strong>:
Many of the tools that we will explore can be select via multiple different routes within the browser interface. One way to access many tools is via from the top tool bar on a pull down list.  In the following instructions a forward slash '/ ' is used to indicate lower levels from the pull down menu on this main tool bar.  For example, the notation below indicates that you should select 'Genome Browser' from the top tool bar and then click on 'Reset all user settings'.</p>
<ul>
<li>Toolbar / Genome Browser / Reset all user settings</li>
</ul>
<p><strong>Accessing help and training</strong>:
This workshop the UCSC genome browser is supported by rich training resource <a href="https://www.youtube.com/channel/UCQnUJepyNOw0p8s2otX4RYQ/videos">youtube channel</a> . To access training to further develop your skills and go to:
Toolbar / Help / Training /</p>
<h3 id="getting-started">Getting started</h3>
<ol>
<li><strong>Open the Browser interface:</strong><ul>
<li>Navigate to the <a href="https://genome.ucsc.edu/">UCSC genome Browser</a> and sign in if you have an account.<br>
<img src="../media/UCSC_Home.png" width="700"></li>
</ul>
</li>
<li><strong>First reset the browser, so that we all see the same screen:</strong><ul>
<li>Toolbar / Genome Browser / Reset all user settings</li>
</ul>
</li>
<li>
<p><strong>Select and open the human Genome Hg38 at the default position, there are a few different ways to do this</strong></p>
<ul>
<li>Toolbar / Genomes (this takes you to the Genome gateway page)<ul>
<li>Ensure that GRCh38 is selected in 'human assembly' box and click on the blue "GO" box</li>
</ul>
</li>
<li>Toolbar / Genomes / Human GRCh38hg38 (takes you directly to the genome)</li>
</ul>
<p>You should see this screen, opening at a position on the X chromosome of Human genome version GRCh38 showing the gene model for the ACE2 gene.
<img src="../media/UCSC_Hg38_opening.png" width="700"></p>
</li>
<li>
<p><strong>Familiarise yourself with the main areas of the interface and locate:</strong></p>
<ul>
<li>The main Toolbar</li>
<li>Blue bar track collections (data of similar types are collected together under the same 'Blue bar' heading). Scroll down to see additional data collections and which ones are turned on as default.</li>
<li>Genome species and version number</li>
<li>Position box</li>
<li>Navigation tool buttons</li>
<li>Chromosome ideogram</li>
<li>Genome view window</li>
<li>Pre loaded tracks, track titles:<ul>
<li>The grey bars on the left of the genome view can be used for selecting and configuring the tracks.</li>
<li>You can change the order of the tracks by moving these grey bars up and down.</li>
</ul>
</li>
<li>Turn tracks on and off:<ul>
<li>You can hide tracks by right clicking on the grey bar or by turning them off in the Blue bar collections.</li>
<li>You have to click on a 'refresh' button to the changes to be reflected in the the genome view window.</li>
</ul>
</li>
</ul>
</li>
<li>
<p><strong>Customise your view by using the 'Configure' tool to change the font size to 12.</strong> Use either method below to open the Configure tool.</p>
<ul>
<li>Toolbar / View / Configure browser / text size 12 / submit</li>
<li>Click on white 'configure' button below the genome view window</li>
<li>Select white 'resize' button to fit the genome view window to your screen</li>
</ul>
</li>
<li><strong>Practice navigating around the genome view</strong><ul>
<li>move left and right both the navigation buttons and your mouse</li>
<li>zoom in and out using navigation buttons</li>
<li>zoom in to a region of interest using 'Drag and select': select a region in the ruler (position track) at the top of the genome view window.</li>
<li>This is also how to access the 'highlight tool' which you will use in a later exercise. Click on the down arrow next to the highlight colour to select a different colour.
<img src="../media/UCSC_drag-and-select.png" width="700"></li>
</ul>
</li>
<li><strong>The top preloaded track in the genome view is the genome patches.</strong>  <ul>
<li>The importance of patches is explained in the <a href="http://genome.ucsc.edu/blog/patches/">UCSC browser blog</a></li>
</ul>
</li>
<li><strong>Find data in the 'Alternate haplotype track'</strong> Since only one haplotype can be represented in the reference genome, highly variable regions of the genome with sets of variants that tend to segregate together are made available as a track aligned with the reference genome.   <ul>
<li>The region of the genome that opens as default view does not contain any data on the haplotype track. We are going to zoom out until you can see some data on this track and then zoom in to that region. But first we must turn off all the other tracks as the more data you are trying to view in the window the busier the view gets and the slower it will run.</li>
<li>Turn off individual tracks by right clicking on the grey bar for the track, or the track title in the genome view, and selecting 'hide'. If you accidentally hide the 'Alt haplotype' track you can turn it on again from the 'Mapping and Sequencing' Blue bar collection. (Don't forget to click on 'refresh' when turning on tracks)</li>
<li>As you are zooming out note the change in the red line in the chromosome ideogram and how many bp are covered in the genome view.</li>
<li>Zoom out so that you are viewing the whole chromosome X in the genome view.<ul>
<li>How many alternate haplotypes are included in the datatrack. Hint: switch between the 'dense' and 'full' view of the track.
<img src="../media/UCSC_trackview.png" width="700"></li>
</ul>
</li>
<li>Zoom in one one of these haplotypes by selecting the region in the number track at the top. then find out what the different colours signify in the colouring scheme by viewing the configuration page specific to this track.  The configuration page gives you a lot of information about the data track and its colouring. You can open the configuration page for a track by:<ul>
<li>clicking on the grey bar for the track or,</li>
<li>clicking on the track title in the Blue bar collection.</li>
</ul>
</li>
<li>More information and options is usually available by selecting the configuration page for a track via the title in the Blue Bar collections.</li>
</ul>
</li>
<li>
<p><strong>Change location in the genome to by typing in the position box.</strong> You can navigate to a different region of the genome by typing in the position box.</p>
<ul>
<li>If you know the specific location you are interested in type in the location using the format "chr#:1234-1234".</li>
<li>If you have a gene of interest you can type in the gene name (eg: HERC2, OCA2). Note the autocompleted suggestions that appear when  you start typing.  You can select from one of the suggestions or click 'go' and select from a wider range of options.<ul>
<li>Which chromosome are these two genes on?</li>
<li>Are they close together?</li>
<li>What strand are they encoded on?<ul>
<li>Use the white 'reverse' button under the genome view to flip the orientation of the gene.</li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
<li>
<p><strong>Loading different tracks for comparison</strong></p>
<ul>
<li>Turn on the OMIM Alleles track (full) in the phenotype and Literature blue bar collection.<ul>
<li>Are there any SNPs that affect phenotype which are also located in intronic regions?</li>
</ul>
</li>
<li>Turn on the 'Conservation' track in 'Comparative Genomics' Blue bar collection. Note that the most conserved regions correspond with coding regions. Zoom into a coding region until you can see the amino acid sequence and the nucleotide bases.  <ul>
<li>Is the amino acid sequence conserved between vertebrates?</li>
<li>Is the conservation level constant across all 3 bases in the codon?</li>
</ul>
</li>
<li>Type the SNP id 'rs3769955' into the location box.<ul>
<li>What gene is this snp in?</li>
<li>Is the SNP associated with any disease?</li>
</ul>
</li>
</ul>
</li>
</ol>
<h3 id="understanding-the-gene-models">Understanding the gene models</h3>
<p><img src="../media/gene_model.png" width="700"><br></p>
<ol>
<li>
<p><strong>Navigate to the NTRK2 gene position in GRCh38 and view the gene models</strong></p>
<ul>
<li>'Hide all' tracks by selecting the white button below the genome view.</li>
<li>Turn on only the 'Genecode v32 Genemodels' in 'full' viewing mode by selecting from the blue bar group labeled 'Genes and Gene predictions'.</li>
<li>Turn on 'Conservation' track to 'full'<ul>
<li>Dont forget to click 'refresh'</li>
</ul>
</li>
<li>Type (or copy and paste) <strong>NTRK2</strong> or <strong>chr9:84,665,760-85,030,334</strong> into the position box.</li>
<li>
<p>Navigate to the NTRK2 gene, zoom out until you can view all of the 5' UTRs and 3' UTRs for all transcript variants for this gene. Then drag the view left and right to center (like in Google maps) or 'drag and select' the region to center the gene in the Genome view.
<img src="../media/UCSC_NTRK2_GENCODE.png" width="700"></p>
</li>
<li>
<p>Which strand is the gene encoded on / transcribed from? (+ or - strand)</p>
</li>
<li>Identify the exons, introns and UTRs</li>
<li>Do regions of conservation only occur around exons?</li>
<li>How many different transcripts variants are there for this gene? How do they differ?</li>
<li>Select a coding region (full height boxes) towards the 3'UTR of the gene<ul>
<li>zoom in until you can see the letters of the amino acid sequence.</li>
<li>Zoom in again until you can see each amino acid number.  </li>
</ul>
</li>
<li>Why do different transcripts have different amino acid numbers?</li>
<li>Note that one of the transcript names is in white text with a black background, this is the transcript you selected from the autocompleted list or the search results.</li>
<li>Change the view settings (dense / squish / pack / full) to see how it changes the representation of the models.</li>
<li>Go to the configuration  page for the Gencode v32 track and change the gene names to also reveal the 'Gencode transcript ID' in the label.</li>
<li>The transcript names are now too long to fit on the screen. Go to the genome view configuration page and change the number of characters in the label so that you can see the entire transcript label.  </li>
<li>Once you think you understand the gene model representation navigate to some of the genes below and for each gene determine which strand the gene is encoded on and transcript variant has:<ul>
<li>the longest 3'UTR?</li>
<li>the longest 3'UTR?</li>
<li>the shortest CDS</li>
<li><strong>TMEM101,
G6PC,
PEX16,
NKAPD1,
ZSCAN29</strong></li>
</ul>
</li>
</ul>
</li>
</ol>
<h3 id="gene-expression-data">Gene expression data</h3>
<p><img src="../media/coverage_plot.png" width="700"></p>
<ol>
<li>
<p><strong>Human tissue specific expression data from the <a href="https://gtexportal.org/home/">GTEX project</a> is available in UCSC genome browser</strong></p>
<ul>
<li>Gene level expression data from GTEx V6 (570) donors, and GTEx V8 (948) donors can be turned on from Blue bar title for more detailed configuration page. These are displayed as coloured bar plots.</li>
<li>Transcript level data is also available for GTEx V6, this is also displayed as bar plots.</li>
<li>Transcript level expression data for GTEx V6 is available as coverage plots and is accessed from the <strong>'Track hubs'</strong>.<ul>
<li>click on the white 'track hubs' under the genome view window or access from the toolbar under 'My data'.</li>
<li>scroll down and select 'GTEx RNA-seq Signal Hub'.  This can be viewed in either hg19 or hg38 so can be compared with a wide variety of other datasets.
<img src="../media/trackhubs_GTEx.png" width="700"></li>
</ul>
</li>
<li>The default settings are:<ul>
<li>that all the available data from one individual is loaded. Other samples from other subjects in the study can be loaded.  For example you could load all available samples for one tissue region only.</li>
<li>the data is 'autoscale to data view' with a track height or 12 pixels for each samples. You can change the height of the track or add a transformation.</li>
<li>you can also change it from a points plot into a density plot by clicking on the track title.
<img src="../media/GTEx_V6_settings.png" width="600"></li>
</ul>
</li>
</ul>
</li>
<li>
<p><strong>The FACS derived data from the <a href="https://tabula-muris.ds.czbiohub.org/">Tabular Muris</a> cell type data can be visualised as a coverage plot</strong></p>
<ul>
<li>Start at the view of the NTRK2 gene in one of the human genomes and navigate to the NTRK2 gene in the mouse genome using the 'View in other genomes tool'.<ul>
<li>Toolbar / View / In Other Genomes (Convert)</li>
<li>select New Genome:Mouse , New Assembly:GRC38/mm10, click on 'Submit'</li>
<li>select the region with the greatest homology</li>
</ul>
</li>
<li>Configure the Tabular Muris track by selecting it from the blue bar collection.<ul>
<li>Hide 'Cell expression'</li>
<li>Select 'Genome coverage' to full</li>
<li>Select 'submit'<br>
<strong>This can look like a bit too much data to manage as there are very many tracks and the default track height is set very high. But its easy to simplify it by focusing on cell types of interest.</strong><br></li>
</ul>
</li>
<li>Right click on the grey bar to 'configure the track set'.<ul>
<li>Change track height to 30</li>
<li>for 'data view scaling' select group autoscale</li>
<li>clear all the subtracks and then manually select only a few cell types of interest:<ul>
<li>astrocyte Cv</li>
<li>Bergmann glial Cv</li>
<li>microglia Cv</li>
<li>neuron Cv</li>
<li>oligodendrocyte Cv</li>
<li>OPC Cv</li>
</ul>
</li>
</ul>
</li>
<li>Which cell type has the highest level expression in this dataset?</li>
<li>Change the 'Data view scaling' to autoscale to dataview.<ul>
<li>Export a PDF image of the genome view: Toolbar / View / PDF/PS / select 'Download the current browser graphic in PDF'
<img src="../media/mm10_Ntrk2.png" width="600"></li>
</ul>
</li>
<li>Which cell type(s) express the long and short transcripts for NTRK2?</li>
</ul>
</li>
</ol>
<h3 id="blat-tool">Blat tool</h3>
<p>The <a href="https://genome.ucsc.edu/FAQ/FAQblat.html">Blat tool</a> is a sequence similarity tool similar to Blast. It can quickly identify region(s) of homology between a genome and a sequence of interest. Due to the presence of <a href="https://en.wikipedia.org/wiki/Sequence_homology">orthologs and paralogs</a> a target sequence may have similarity to more than one region in the genome. In this exercise you will use Blat to map the sequences of two different expression probes to their target regions and determine which gene transcripts the probes are likely to detect in an expression study.</p>
<p>Below are sequences of two hybridisation probes that were use in a microarray used to detect expression of the gene NTRK2. The study was <a href="http://human.brain-map.org/">the Human Brain gene expression atlas</a> generated by the <a href="https://alleninstitute.org/about/">Allen Institute</a>.  These two probes result in very different hybridisation and expression patterns across different regions of the brain. As we observed in the exercise above NTRK2 has a number of different transcript variants.  The question is whether these probe sets are detecting different transcripts of NTRK2, and if so which ones?</p>
<p><strong><em>NTRK2 Probe A_23_P216779 sequence:</em></strong></br>
<strong>TTCTATACTCTAATCAGCACTGAATTCAGAGGGTTTGACTTTTTCATCTATAACACAGTG</strong></br>
Z score of expression level in Human brain (blue = low expression, red = high expression)
<img src="../media/ABA_P216779.png" width="700"></p>
<p><strong><em>NTRK2 Probe A_24_P343559 sequence</em></strong></br>
<strong>AAGCTGCTCTCCTTCACTCTGACAGTATTAACATCAAAGACTCCGAGAAGCTCTCGAGGG</strong></br>
Z score of expression level in Human brain (blue = low expression, red = high expression)
<img src="../media/ABA_P343559.png" width="700"></p>
<p><strong><em>The images above are of one of the six donors included in the atlas, and typical of the expression pattern for NTRK2. These images are taken from the <a href="http://human.brain-map.org/microarray/gene/show/4884">NTRK2 gene page of Human Brain Atlas</a>.</em></strong></p>
<p>Most obvious in the images above is the high level of expression signal using Probe A_23_P216779 and low level for A_24_P343559 in the corpus callosum (CC) which is a region of white matter in the brain with relatively few neurons. This expression profile is reversed in the the cortical regions, eg. frontal lobe (FL) and parietal lobe (PL), which have a relatively high density of neuronal cells.</p>
<ol>
<li>
<p><strong>Use Blat tool to find region of homology</strong></p>
<ul>
<li>Select: Toolbar / tools/ blat</li>
<li>Copy the sequence of the first probe above and paste into the search box</li>
<li>Select the human GRCh38  for 'Assembly' and click 'Submit'<ul>
<li>A_23_P216779 returns 2 hits for different chromosomes.  One of these has 100% homology over the whole 60 base sequence, the other has 87% homology over a 24 base region.</li>
</ul>
</li>
<li>Copy and paste the probe name as the label for the Custom track name and description and select 'Build a custom track with these results'. It's not necessary to build a custom track, you could simply click on the 'browser' link to view the results, but creating a custom track from the blat result enables you to give it a unique name.  This is important if you are doing multiple blat searches.</li>
<li>Select ‘browser’ option for the hit with the highest homology to view the result.</li>
<li>repeat for the other probe sequence.</li>
</ul>
</li>
<li>
<p><strong>Use 'highlight' tool to keep track of region of interest in the Genome view</strong> It is easy to loose track of a region you are investigating when zooming out or scrolling down.  </p>
<ul>
<li>Select only the region of homology for each probe within the NTRK2 gene and use a different highlight for each region. Then zoom out to view the whole gene again.</li>
<li>Do the probes detect coding regions of the NTRK2 gene?</li>
<li>Do the probes detect different transcripts?</li>
</ul>
</li>
<li>
<p><strong>Use 'Multiregion view' to make it easier to compare coding regions of different transcripts</strong></p>
<ul>
<li>Toolbar / View / Multi-region View / Show exons using Gencode v32. Use padding of 6 bases.</li>
</ul>
</li>
</ol>
<p><strong>I have created a 'public session' of the Blat NTRK2 exercise you can view it from the link in the sessions</strong></br>
    * Toolbar / My data / Public session / search for "hg38_NTRK2_blat_probes"</p>
<h2 id="section-3-igv">Section 3: IGV</h2>
<p>In this section we will download a BAM file of gene expression data from SRA and view it in the Integrated Genome Viewer (IGV).  BAM files must first be sorted and indexed before they can be loaded into genome viewers and IGV has tools to do this without having to use command line.</p>
<p>The express ion data we are using for this exercise is from the mouse <a href="http://casestudies.brain-map.org/celltax">Celltax single cell expression atlas</a> published by the Allen Brain Institute.  The cell tax vignette has an expression browser that displays gene level expression as a heat map for any gene of interest,  The readsets (fastq files) and aligned data (BAM files) for 1809 runs on single cells are also available for down load from SRA.</p>
<p>The SRA study ID for this study is <a href="https://www.ncbi.nlm.nih.gov/sra?term=SRP061902">SRP061902</a> and individual runs from this study are easily selected by viewing the samples in the 'RunSelector'. If you wish to identify particular cell types of interest.  For this exercise I have already identified a few samples that we will download in order to illustrate navigating in IGV by looking at the expression of NTRK2 in the same cell types we have discussed in earlier exercises.<br></p>
<p>For each cell type we will down load a .BAM file containing only the reads from the chromosome of interest.</p>
<p>For each SRA run in the table below open the link to the run to down load the data. Not many raw data sets in SRA have aligned data available for down load but this data set does.</p>
<table>
<thead>
<tr>
<th align="left">Cell type</th>
<th align="left">SRA run</th>
<th align="left">Vignette Cell ID</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left"><strong>astrocyte</strong></td>
<td align="left"><a href="https://trace.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?run=SRR2138962">SRR2138962</a></td>
<td align="left">D1319_V</td>
</tr>
<tr>
<td align="left"><strong>astrocyte</strong></td>
<td align="left"><a href="https://trace.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?run=SRR2139935">SRR2139935</a></td>
<td align="left">A1643_VL</td>
</tr>
<tr>
<td align="left"><strong>neuron</strong></td>
<td align="left"><a href="https://trace.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?run=SRR2139989">SRR2139989</a></td>
<td align="left">S467_V4</td>
</tr>
<tr>
<td align="left"><strong>neuron</strong></td>
<td align="left"><a href="https://trace.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?run=SRR2140047">SRR2140047</a></td>
<td align="left">S1282_V</td>
</tr>
</tbody>
</table>
<ol>
<li>
<p><strong>Download BAM files from SRA</strong></p>
<ul>
<li>Click on the 'Alignment' tab</li>
<li>Note that the data is aligned to the mouse GRCm38 genome (mm10).</li>
<li>Select the chromosome of interest. For NTRK2 in mouse it is chr13</li>
<li>For 'Output this run in:' select <strong>BAM</strong> and click on 'format to:' <strong>File</strong></li>
<li>Rename the downloaded file to include the cell type, to avoid confusion. eg: <strong>SRR2138661_astrocyte_chr13.bam</strong></li>
</ul>
</li>
<li>
<p><strong>Use IGV tools to SORT and INDEX the BAM files</strong> Store sorted BAM files and index files in the same folder.</p>
<ul>
<li>Open IGV and select Tools / Run igvtools... from the pull town menus.</li>
<li>Select <strong>'Sort'</strong> from the Command options and use the brows options to select the BAM file you just downloaded and click <strong>'Run'</strong></li>
<li>Without closing the igvtools window now select the command <strong>'Index'</strong> and Browse to find the BAM file you just sorted.  It will have the same file name with 'sorted' added to the end. eg SRR2138661_astrocyte_chr13.sorted.bam</li>
<li>The resulting index file will have the file name : SRR2138661_astrocyte_chr13.sorted.bam.bai<ul>
<li>It is essential that the index file for a BAM file has the same name and is located in the same folder as its BAM file. If not the IVG software will not be able to open the BAM file.<br></li>
</ul>
</li>
</ul>
</li>
</ol>
<p><img src="../media/BAM_bai_files.png" width="300"></p>
<ol>
<li><strong>View the BAM files in IGV</strong><ul>
<li>Select the Mouse (mm10) genome from the genome box in the top right hand corner.</li>
<li>Select File / Load from File... and select all 4 '_chr13.sorted.bam' files only (use command to select more then one file at a time).</li>
<li>select open - but don't expect to see any data yet. The genome view window opens on a whole chromosome view as default but it wont show any data until the view region is small enough to show all data in the current view.</li>
<li>Type the gene name 'NTRK2' into the search window.</li>
<li>Expand the Refseq gene model track by right clicking it to see all the splice variants</li>
<li>The gene and thus the genome view is 328kb and the default setting for viewing data is only 100kb.  So unless you have already changed your settings alignment data will not get be showing.</li>
<li>zoom into the region of a coding exon by selecting in the numbered location track at the top of the genome view.</li>
<li>To see the whole gene in the genome window at the same time you may need to change the preferences.  </li>
<li>Go to View / Preferences and select the 'Alignments tab'.</li>
<li>Change the visibility range threshold to 400kb.</li>
</ul>
</li>
</ol>
<p><img src="../media/IGV_preferences.png" width="400"></p>
<p><strong>You may need to change this back to a smaller range in the future if you are working with large datasets and/or small amounts of memory on your computer.</strong></p>
<p><img src="../media/igv_Ntrk2.png" width="700"></p>
<ol>
<li>
<p><strong>Export images</strong></p>
<ul>
<li>The Genome view above can be exported by selecting 'File / Save image...'  from the tool bar.</li>
<li>To export the Sashimi plot below:<ul>
<li>Right click on one of the junction tracks and select 'Sashimi Plot' from the poll down menu.</li>
<li>Select the tracks you want in your final image.</li>
<li>There are some data filtering and style adjustments you can make to the Sashimi plot. Right click on each track to access the menu options. Some changes apply to each track individually and some to all tracks.
    *
<img src="../media/Sashimi.png" width="700"></li>
</ul>
</li>
</ul>
</li>
<li>
<p><strong>Download and install the Gencode gene model annotation track</strong></p>
<ul>
<li>The refseq gene model track is not as comprehensive as GeneGode gene models. For both Human and Mouse the Gencode gene model gtf annotaion files can be downloaded form <a href="https://www.gencodegenes.org/mouse/">Gencode</a>. If you wish to do this be aware that it takes a little time and is not  done as part of a workshop.<ul>
<li>Create a folder called 'annotations/Mouse' in the main 'igv' folder that was installed on your computer when you downloaded IGV.</li>
<li>Download the GTF file from the link above and save it in this folder.</li>
<li>Unpack and then <strong>SORT</strong> and <strong>INDEX</strong> the .gtf file using igvtools.</li>
<li>in IGV, before you load you data files, load this annotation file and it will replace the refseq one.</li>
</ul>
</li>
</ul>
</li>
</ol>
<hr />
<h2 id="additional-reading">Additional reading</h2>
<p><strong>IGV</strong>
https://rockefelleruniversity.github.io/IGV_course/presentations/singlepage/IGV.html</p>
                
                  
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../../unix/unix/" title="Introduction to Unix" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                Introduction to Unix
              </span>
            </div>
          </a>
        
        
          <a href="../../rna_seq_exp_design/rna_seq_experimental_design/" title="RNA-seq Experimental Design" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                RNA-seq Experimental Design
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            <a href="https://www.melbournebioinformatics.org.au/">Melbourne Bioinformatics</a>
          </div>
        
        powered by
        <a href="https://www.mkdocs.org">MkDocs</a>
        and
        <a href="https://squidfunk.github.io/mkdocs-material/">
          Material for MkDocs</a>
      </div>
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../../assets/javascripts/application.245445c6.js"></script>
      
      <script>app.initialize({version:"1.0.4",url:{base:"../../.."}})</script>
      
    
  </body>
</html>